{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "XUZS4GUkHIEG"
      },
      "source": [
        "# Подготовка файлов и программ"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 1,
      "metadata": {
        "id": "lrtRbP3fAvj8",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "37c2243d-e9c2-46af-a6bc-9df294a0b6c8"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\u001b[33m\n",
            "\u001b[0m"
          ]
        }
      ],
      "source": [
        "!pip install -q condacolab"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 2,
      "metadata": {
        "id": "jQ1Oqzb9BXoG"
      },
      "outputs": [],
      "source": [
        "import condacolab"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 3,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "q5FUOHCgBZWp",
        "outputId": "ad099b9a-8ae3-443a-f1b9-92830f3e258f"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "✨🍰✨ Everything looks OK!\n"
          ]
        }
      ],
      "source": [
        "condacolab.install()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 4,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "XnbIt77YBOT9",
        "outputId": "632dc207-46ad-4532-c45a-11d2b2c84a66"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
            "Solving environment: | \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
            "\n",
            "# All requested packages already installed.\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!conda install -c bioconda seqtk"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 5,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "sCBV_31qHOik",
        "outputId": "39dcf8a3-208c-4bb9-d7e7-b704e9a02d57"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
            "Solving environment: / \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
            "\n",
            "# All requested packages already installed.\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!conda install -c bioconda trimmomatic"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 54,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "F0BLWsH2n3Ov",
        "outputId": "086c9f29-b3cc-45ce-dfed-d30db44baddf"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/\n",
            "Requirement already satisfied: intervene in /usr/local/lib/python3.8/site-packages (0.6.5)\n",
            "Requirement already satisfied: numpy in /usr/local/lib/python3.8/site-packages (from intervene) (1.24.2)\n",
            "Requirement already satisfied: seaborn in /usr/local/lib/python3.8/site-packages (from intervene) (0.12.2)\n",
            "Requirement already satisfied: pybedtools in /usr/local/lib/python3.8/site-packages (from intervene) (0.9.0)\n",
            "Requirement already satisfied: matplotlib in /usr/local/lib/python3.8/site-packages (from intervene) (3.7.1)\n",
            "Requirement already satisfied: scipy in /usr/local/lib/python3.8/site-packages (from intervene) (1.10.1)\n",
            "Requirement already satisfied: pandas in /usr/local/lib/python3.8/site-packages (from intervene) (1.5.3)\n",
            "Requirement already satisfied: importlib-resources>=3.2.0 in /usr/local/lib/python3.8/site-packages (from matplotlib->intervene) (5.12.0)\n",
            "Requirement already satisfied: pillow>=6.2.0 in /usr/local/lib/python3.8/site-packages (from matplotlib->intervene) (9.4.0)\n",
            "Requirement already satisfied: pyparsing>=2.3.1 in /usr/local/lib/python3.8/site-packages (from matplotlib->intervene) (3.0.9)\n",
            "Requirement already satisfied: packaging>=20.0 in /usr/local/lib/python3.8/site-packages (from matplotlib->intervene) (23.0)\n",
            "Requirement already satisfied: python-dateutil>=2.7 in /usr/local/lib/python3.8/site-packages (from matplotlib->intervene) (2.8.2)\n",
            "Requirement already satisfied: cycler>=0.10 in /usr/local/lib/python3.8/site-packages (from matplotlib->intervene) (0.11.0)\n",
            "Requirement already satisfied: fonttools>=4.22.0 in /usr/local/lib/python3.8/site-packages (from matplotlib->intervene) (4.39.0)\n",
            "Requirement already satisfied: kiwisolver>=1.0.1 in /usr/local/lib/python3.8/site-packages (from matplotlib->intervene) (1.4.4)\n",
            "Requirement already satisfied: contourpy>=1.0.1 in /usr/local/lib/python3.8/site-packages (from matplotlib->intervene) (1.0.7)\n",
            "Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.8/site-packages (from pandas->intervene) (2022.7.1)\n",
            "Requirement already satisfied: pysam in /usr/local/lib/python3.8/site-packages (from pybedtools->intervene) (0.20.0)\n",
            "Requirement already satisfied: six in /usr/local/lib/python3.8/site-packages (from pybedtools->intervene) (1.16.0)\n",
            "Requirement already satisfied: zipp>=3.1.0 in /usr/local/lib/python3.8/site-packages (from importlib-resources>=3.2.0->matplotlib->intervene) (3.15.0)\n",
            "\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\u001b[33m\n",
            "\u001b[0m"
          ]
        }
      ],
      "source": [
        "!pip install intervene"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!pwd"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "qDSUd3R6qB-d",
        "outputId": "5ef2fd86-4011-44a2-ebdb-7e39ad11f81b"
      },
      "execution_count": 60,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "/content\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# !wget https://github.com/macs3-project/MACS/archive/refs/tags/v2.2.7.1.tar.gz -O MACS.tar.gz\n",
        "# !tar -xvf MACS.tar.gz\n",
        "!cd /content/MACS-2.2.7.1\n",
        "!sed -i 's/install_requires = \\[f\"numpy>={numpy_requires}\",\\]/install_requires = \\[f\"numpy{numpy_requires}\",\\]/' /content/MACS-2.2.7.1/setup.py\n",
        "!cd /content/MACS-2.2.7.1\n",
        "!pip install -e /content/MACS-2.2.7.1"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "F_GiTpO2pCBC",
        "outputId": "8f1cb18a-5d9f-4984-815e-4e40d53a7608"
      },
      "execution_count": 69,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/\n",
            "Obtaining file:///content/MACS-2.2.7.1\n",
            "  Installing build dependencies ... \u001b[?25l\u001b[?25hdone\n",
            "  Checking if build backend supports build_editable ... \u001b[?25l\u001b[?25hdone\n",
            "  Getting requirements to build editable ... \u001b[?25l\u001b[?25hdone\n",
            "  Preparing editable metadata (pyproject.toml) ... \u001b[?25l\u001b[?25hdone\n",
            "Requirement already satisfied: numpy>=1.17 in /usr/local/lib/python3.8/site-packages (from MACS2==2.2.7.1) (1.24.2)\n",
            "Building wheels for collected packages: MACS2\n",
            "  Building editable for MACS2 (pyproject.toml) ... \u001b[?25l\u001b[?25hdone\n",
            "  Created wheel for MACS2: filename=MACS2-2.2.7.1-0.editable-cp38-cp38-linux_x86_64.whl size=26960 sha256=6aab209e684e91946365bf77dfda89be27abe5d86ecc37d0f078ec0a3b3ddd91\n",
            "  Stored in directory: /tmp/pip-ephem-wheel-cache-9b5xguz1/wheels/16/eb/be/f56e2021c4aaf34ff68e097a8d5e200336fd6ba1797d855ae4\n",
            "Successfully built MACS2\n",
            "Installing collected packages: MACS2\n",
            "Successfully installed MACS2-2.2.7.1\n",
            "\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\u001b[33m\n",
            "\u001b[0m"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!pip install /content/MACS-2.2.7.1/pyproject.toml"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "dSBmf5Htq3l5",
        "outputId": "4d6bd0b7-ff4f-4407-eca3-816017ae4610"
      },
      "execution_count": 68,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\u001b[31mERROR: Invalid requirement: '/content/MACS-2.2.7.1/pyproject.toml'\n",
            "Hint: It looks like a path. The path does exist. \u001b[0m\u001b[31m\n",
            "\u001b[0m"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 7,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Ce6hiSTdkarU",
        "outputId": "69db7544-07b3-47c0-8ba4-4c599a2139c3"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "bedtools is already the newest version (2.27.1+dfsg-4ubuntu1).\n",
            "0 upgraded, 0 newly installed, 0 to remove and 22 not upgraded.\n"
          ]
        }
      ],
      "source": [
        "!apt-get install bedtools"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 8,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "AUML3dvsqT9E",
        "outputId": "31c87690-660e-48a7-9278-3c78cc53a1f4"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "The following additional packages will be installed:\n",
            "  cwltool fonts-glyphicons-halflings libc-ares2 libhts3 libjs-bootstrap\n",
            "  libnode64 nodejs nodejs-doc python3-bs4 python3-cachecontrol\n",
            "  python3-coloredlogs python3-dateutil python3-decorator python3-html5lib\n",
            "  python3-humanfriendly python3-isodate python3-lockfile python3-lxml\n",
            "  python3-mistune python3-msgpack python3-mypy-extensions python3-networkx\n",
            "  python3-prov python3-psutil python3-pyparsing python3-rdflib\n",
            "  python3-rdflib-jsonld python3-ruamel.yaml python3-schema-salad\n",
            "  python3-shellescape python3-soupsieve python3-sparqlwrapper\n",
            "  python3-typing-extensions python3-webencodings python3-yaml\n",
            "Suggested packages:\n",
            "  bowtie2-examples docker.io | singularity-container npm python3-genshi\n",
            "  python-lockfile-doc python3-lxml-dbg python-lxml-doc python-networkx-doc\n",
            "  python3-matplotlib python3-pygraphviz | python3-pydot python3-scipy\n",
            "  python-prov-doc python3-pydot python-psutil-doc python-pyparsing-doc\n",
            "  python-rdflib-doc\n",
            "The following NEW packages will be installed:\n",
            "  bowtie2 cwltool fonts-glyphicons-halflings libc-ares2 libhts3\n",
            "  libjs-bootstrap libnode64 nodejs nodejs-doc python3-bs4 python3-cachecontrol\n",
            "  python3-coloredlogs python3-dateutil python3-decorator python3-html5lib\n",
            "  python3-humanfriendly python3-isodate python3-lockfile python3-lxml\n",
            "  python3-mistune python3-msgpack python3-mypy-extensions python3-networkx\n",
            "  python3-prov python3-psutil python3-pyparsing python3-rdflib\n",
            "  python3-rdflib-jsonld python3-ruamel.yaml python3-schema-salad\n",
            "  python3-shellescape python3-soupsieve python3-sparqlwrapper\n",
            "  python3-typing-extensions python3-webencodings python3-yaml samtools\n",
            "0 upgraded, 37 newly installed, 0 to remove and 22 not upgraded.\n",
            "Need to get 13.4 MB of archives.\n",
            "After this operation, 64.0 MB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 python3-yaml amd64 5.3.1-1ubuntu0.1 [136 kB]\n",
            "Get:2 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-humanfriendly all 4.18-2 [60.8 kB]\n",
            "Get:3 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-coloredlogs all 7.3-2 [28.0 kB]\n",
            "Get:4 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-mypy-extensions all 0.4.3-1 [5,436 B]\n",
            "Get:5 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-dateutil all 2.7.3-3ubuntu1 [63.3 kB]\n",
            "Get:6 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 python3-lxml amd64 4.5.0-1ubuntu0.5 [1,384 kB]\n",
            "Get:7 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-decorator all 4.4.2-0ubuntu1 [10.3 kB]\n",
            "Get:8 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-networkx all 2.4-3ubuntu1 [855 kB]\n",
            "Get:9 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-isodate all 0.6.0-2 [24.0 kB]\n",
            "Get:10 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-pyparsing all 2.4.6-1 [61.3 kB]\n",
            "Get:11 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-rdflib all 4.2.2-5 [241 kB]\n",
            "Get:12 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-prov all 1.5.2-2 [92.8 kB]\n",
            "Get:13 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-psutil amd64 5.5.1-1ubuntu4 [141 kB]\n",
            "Get:14 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-ruamel.yaml amd64 0.15.89-3build1 [228 kB]\n",
            "Get:15 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-msgpack amd64 0.6.2-1 [73.5 kB]\n",
            "Get:16 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-lockfile all 1:0.12.2-2ubuntu2 [14.6 kB]\n",
            "Get:17 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-cachecontrol all 0.12.6-1ubuntu2 [16.7 kB]\n",
            "Get:18 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-mistune all 0.8.4-2 [14.3 kB]\n",
            "Get:19 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-rdflib-jsonld all 0.4.0-5 [12.4 kB]\n",
            "Get:20 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-typing-extensions all 3.7.4.1-1 [27.5 kB]\n",
            "Get:21 http://archive.ubuntu.com/ubuntu focal/universe amd64 fonts-glyphicons-halflings all 1.009~3.4.1+dfsg-1 [117 kB]\n",
            "Get:22 http://archive.ubuntu.com/ubuntu focal/universe amd64 libjs-bootstrap all 3.4.1+dfsg-1 [124 kB]\n",
            "Get:23 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-schema-salad all 5.0.20200220195218-1 [295 kB]\n",
            "Get:24 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-shellescape all 3.4.1-4 [4,928 B]\n",
            "Get:25 http://archive.ubuntu.com/ubuntu focal/universe amd64 cwltool all 2.0.20200224214940+dfsg-1 [203 kB]\n",
            "Get:26 http://archive.ubuntu.com/ubuntu focal-updates/universe amd64 libhts3 amd64 1.10.2-3ubuntu0.1 [350 kB]\n",
            "Get:27 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 libc-ares2 amd64 1.15.0-1ubuntu0.2 [36.7 kB]\n",
            "Get:28 http://archive.ubuntu.com/ubuntu focal/universe amd64 libnode64 amd64 10.19.0~dfsg-3ubuntu1 [5,765 kB]\n",
            "Get:29 http://archive.ubuntu.com/ubuntu focal/universe amd64 nodejs-doc all 10.19.0~dfsg-3ubuntu1 [942 kB]\n",
            "Get:30 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-soupsieve all 1.9.5+dfsg-1 [29.1 kB]\n",
            "Get:31 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-bs4 all 4.8.2-1 [83.0 kB]\n",
            "Get:32 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-webencodings all 0.5.1-1ubuntu1 [11.0 kB]\n",
            "Get:33 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-html5lib all 1.0.1-2 [84.3 kB]\n",
            "Get:34 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-sparqlwrapper all 1.8.5-1 [32.3 kB]\n",
            "Get:35 http://archive.ubuntu.com/ubuntu focal/universe amd64 samtools amd64 1.10-3 [496 kB]\n",
            "Get:36 http://archive.ubuntu.com/ubuntu focal/universe amd64 bowtie2 amd64 2.3.5.1-6build1 [1,324 kB]\n",
            "Get:37 http://archive.ubuntu.com/ubuntu focal/universe amd64 nodejs amd64 10.19.0~dfsg-3ubuntu1 [61.1 kB]\n",
            "Fetched 13.4 MB in 8s (1,682 kB/s)\n",
            "Extracting templates from packages: 100%\n",
            "Selecting previously unselected package python3-yaml.\n",
            "(Reading database ... 128268 files and directories currently installed.)\n",
            "Preparing to unpack .../00-python3-yaml_5.3.1-1ubuntu0.1_amd64.deb ...\n",
            "Unpacking python3-yaml (5.3.1-1ubuntu0.1) ...\n",
            "Selecting previously unselected package python3-humanfriendly.\n",
            "Preparing to unpack .../01-python3-humanfriendly_4.18-2_all.deb ...\n",
            "Unpacking python3-humanfriendly (4.18-2) ...\n",
            "Selecting previously unselected package python3-coloredlogs.\n",
            "Preparing to unpack .../02-python3-coloredlogs_7.3-2_all.deb ...\n",
            "Unpacking python3-coloredlogs (7.3-2) ...\n",
            "Selecting previously unselected package python3-mypy-extensions.\n",
            "Preparing to unpack .../03-python3-mypy-extensions_0.4.3-1_all.deb ...\n",
            "Unpacking python3-mypy-extensions (0.4.3-1) ...\n",
            "Selecting previously unselected package python3-dateutil.\n",
            "Preparing to unpack .../04-python3-dateutil_2.7.3-3ubuntu1_all.deb ...\n",
            "Unpacking python3-dateutil (2.7.3-3ubuntu1) ...\n",
            "Selecting previously unselected package python3-lxml:amd64.\n",
            "Preparing to unpack .../05-python3-lxml_4.5.0-1ubuntu0.5_amd64.deb ...\n",
            "Unpacking python3-lxml:amd64 (4.5.0-1ubuntu0.5) ...\n",
            "Selecting previously unselected package python3-decorator.\n",
            "Preparing to unpack .../06-python3-decorator_4.4.2-0ubuntu1_all.deb ...\n",
            "Unpacking python3-decorator (4.4.2-0ubuntu1) ...\n",
            "Selecting previously unselected package python3-networkx.\n",
            "Preparing to unpack .../07-python3-networkx_2.4-3ubuntu1_all.deb ...\n",
            "Unpacking python3-networkx (2.4-3ubuntu1) ...\n",
            "Selecting previously unselected package python3-isodate.\n",
            "Preparing to unpack .../08-python3-isodate_0.6.0-2_all.deb ...\n",
            "Unpacking python3-isodate (0.6.0-2) ...\n",
            "Selecting previously unselected package python3-pyparsing.\n",
            "Preparing to unpack .../09-python3-pyparsing_2.4.6-1_all.deb ...\n",
            "Unpacking python3-pyparsing (2.4.6-1) ...\n",
            "Selecting previously unselected package python3-rdflib.\n",
            "Preparing to unpack .../10-python3-rdflib_4.2.2-5_all.deb ...\n",
            "Unpacking python3-rdflib (4.2.2-5) ...\n",
            "Selecting previously unselected package python3-prov.\n",
            "Preparing to unpack .../11-python3-prov_1.5.2-2_all.deb ...\n",
            "Unpacking python3-prov (1.5.2-2) ...\n",
            "Selecting previously unselected package python3-psutil.\n",
            "Preparing to unpack .../12-python3-psutil_5.5.1-1ubuntu4_amd64.deb ...\n",
            "Unpacking python3-psutil (5.5.1-1ubuntu4) ...\n",
            "Selecting previously unselected package python3-ruamel.yaml.\n",
            "Preparing to unpack .../13-python3-ruamel.yaml_0.15.89-3build1_amd64.deb ...\n",
            "Unpacking python3-ruamel.yaml (0.15.89-3build1) ...\n",
            "Selecting previously unselected package python3-msgpack.\n",
            "Preparing to unpack .../14-python3-msgpack_0.6.2-1_amd64.deb ...\n",
            "Unpacking python3-msgpack (0.6.2-1) ...\n",
            "Selecting previously unselected package python3-lockfile.\n",
            "Preparing to unpack .../15-python3-lockfile_1%3a0.12.2-2ubuntu2_all.deb ...\n",
            "Unpacking python3-lockfile (1:0.12.2-2ubuntu2) ...\n",
            "Selecting previously unselected package python3-cachecontrol.\n",
            "Preparing to unpack .../16-python3-cachecontrol_0.12.6-1ubuntu2_all.deb ...\n",
            "Unpacking python3-cachecontrol (0.12.6-1ubuntu2) ...\n",
            "Selecting previously unselected package python3-mistune.\n",
            "Preparing to unpack .../17-python3-mistune_0.8.4-2_all.deb ...\n",
            "Unpacking python3-mistune (0.8.4-2) ...\n",
            "Selecting previously unselected package python3-rdflib-jsonld.\n",
            "Preparing to unpack .../18-python3-rdflib-jsonld_0.4.0-5_all.deb ...\n",
            "Unpacking python3-rdflib-jsonld (0.4.0-5) ...\n",
            "Selecting previously unselected package python3-typing-extensions.\n",
            "Preparing to unpack .../19-python3-typing-extensions_3.7.4.1-1_all.deb ...\n",
            "Unpacking python3-typing-extensions (3.7.4.1-1) ...\n",
            "Selecting previously unselected package fonts-glyphicons-halflings.\n",
            "Preparing to unpack .../20-fonts-glyphicons-halflings_1.009~3.4.1+dfsg-1_all.deb ...\n",
            "Unpacking fonts-glyphicons-halflings (1.009~3.4.1+dfsg-1) ...\n",
            "Selecting previously unselected package libjs-bootstrap.\n",
            "Preparing to unpack .../21-libjs-bootstrap_3.4.1+dfsg-1_all.deb ...\n",
            "Unpacking libjs-bootstrap (3.4.1+dfsg-1) ...\n",
            "Selecting previously unselected package python3-schema-salad.\n",
            "Preparing to unpack .../22-python3-schema-salad_5.0.20200220195218-1_all.deb ...\n",
            "Unpacking python3-schema-salad (5.0.20200220195218-1) ...\n",
            "Selecting previously unselected package python3-shellescape.\n",
            "Preparing to unpack .../23-python3-shellescape_3.4.1-4_all.deb ...\n",
            "Unpacking python3-shellescape (3.4.1-4) ...\n",
            "Selecting previously unselected package cwltool.\n",
            "Preparing to unpack .../24-cwltool_2.0.20200224214940+dfsg-1_all.deb ...\n",
            "Unpacking cwltool (2.0.20200224214940+dfsg-1) ...\n",
            "Selecting previously unselected package libhts3:amd64.\n",
            "Preparing to unpack .../25-libhts3_1.10.2-3ubuntu0.1_amd64.deb ...\n",
            "Unpacking libhts3:amd64 (1.10.2-3ubuntu0.1) ...\n",
            "Selecting previously unselected package libc-ares2:amd64.\n",
            "Preparing to unpack .../26-libc-ares2_1.15.0-1ubuntu0.2_amd64.deb ...\n",
            "Unpacking libc-ares2:amd64 (1.15.0-1ubuntu0.2) ...\n",
            "Selecting previously unselected package libnode64:amd64.\n",
            "Preparing to unpack .../27-libnode64_10.19.0~dfsg-3ubuntu1_amd64.deb ...\n",
            "Unpacking libnode64:amd64 (10.19.0~dfsg-3ubuntu1) ...\n",
            "Selecting previously unselected package nodejs-doc.\n",
            "Preparing to unpack .../28-nodejs-doc_10.19.0~dfsg-3ubuntu1_all.deb ...\n",
            "Unpacking nodejs-doc (10.19.0~dfsg-3ubuntu1) ...\n",
            "Selecting previously unselected package python3-soupsieve.\n",
            "Preparing to unpack .../29-python3-soupsieve_1.9.5+dfsg-1_all.deb ...\n",
            "Unpacking python3-soupsieve (1.9.5+dfsg-1) ...\n",
            "Selecting previously unselected package python3-bs4.\n",
            "Preparing to unpack .../30-python3-bs4_4.8.2-1_all.deb ...\n",
            "Unpacking python3-bs4 (4.8.2-1) ...\n",
            "Selecting previously unselected package python3-webencodings.\n",
            "Preparing to unpack .../31-python3-webencodings_0.5.1-1ubuntu1_all.deb ...\n",
            "Unpacking python3-webencodings (0.5.1-1ubuntu1) ...\n",
            "Selecting previously unselected package python3-html5lib.\n",
            "Preparing to unpack .../32-python3-html5lib_1.0.1-2_all.deb ...\n",
            "Unpacking python3-html5lib (1.0.1-2) ...\n",
            "Selecting previously unselected package python3-sparqlwrapper.\n",
            "Preparing to unpack .../33-python3-sparqlwrapper_1.8.5-1_all.deb ...\n",
            "Unpacking python3-sparqlwrapper (1.8.5-1) ...\n",
            "Selecting previously unselected package samtools.\n",
            "Preparing to unpack .../34-samtools_1.10-3_amd64.deb ...\n",
            "Unpacking samtools (1.10-3) ...\n",
            "Selecting previously unselected package bowtie2.\n",
            "Preparing to unpack .../35-bowtie2_2.3.5.1-6build1_amd64.deb ...\n",
            "Unpacking bowtie2 (2.3.5.1-6build1) ...\n",
            "Selecting previously unselected package nodejs.\n",
            "Preparing to unpack .../36-nodejs_10.19.0~dfsg-3ubuntu1_amd64.deb ...\n",
            "Unpacking nodejs (10.19.0~dfsg-3ubuntu1) ...\n",
            "Setting up python3-lockfile (1:0.12.2-2ubuntu2) ...\n",
            "Setting up libhts3:amd64 (1.10.2-3ubuntu0.1) ...\n",
            "Setting up fonts-glyphicons-halflings (1.009~3.4.1+dfsg-1) ...\n",
            "Setting up python3-mypy-extensions (0.4.3-1) ...\n",
            "Setting up python3-yaml (5.3.1-1ubuntu0.1) ...\n",
            "Setting up libc-ares2:amd64 (1.15.0-1ubuntu0.2) ...\n",
            "Setting up python3-webencodings (0.5.1-1ubuntu1) ...\n",
            "Setting up python3-psutil (5.5.1-1ubuntu4) ...\n",
            "Setting up samtools (1.10-3) ...\n",
            "Setting up bowtie2 (2.3.5.1-6build1) ...\n",
            "Setting up python3-isodate (0.6.0-2) ...\n",
            "Setting up python3-decorator (4.4.2-0ubuntu1) ...\n",
            "Setting up python3-pyparsing (2.4.6-1) ...\n",
            "Setting up python3-humanfriendly (4.18-2) ...\n",
            "Setting up libnode64:amd64 (10.19.0~dfsg-3ubuntu1) ...\n",
            "Setting up python3-typing-extensions (3.7.4.1-1) ...\n",
            "Setting up python3-html5lib (1.0.1-2) ...\n",
            "Setting up nodejs-doc (10.19.0~dfsg-3ubuntu1) ...\n",
            "Setting up python3-ruamel.yaml (0.15.89-3build1) ...\n",
            "Setting up python3-lxml:amd64 (4.5.0-1ubuntu0.5) ...\n",
            "Setting up python3-rdflib (4.2.2-5) ...\n",
            "/usr/lib/python3/dist-packages/rdflib/plugins/memory.py:354: SyntaxWarning: \"is\" with a literal. Did you mean \"==\"?\n",
            "  if triple is None or triple is (None,None,None):\n",
            "Setting up python3-dateutil (2.7.3-3ubuntu1) ...\n",
            "Setting up python3-msgpack (0.6.2-1) ...\n",
            "Setting up python3-cachecontrol (0.12.6-1ubuntu2) ...\n",
            "Setting up python3-shellescape (3.4.1-4) ...\n",
            "Setting up python3-coloredlogs (7.3-2) ...\n",
            "Setting up python3-mistune (0.8.4-2) ...\n",
            "Setting up python3-soupsieve (1.9.5+dfsg-1) ...\n",
            "Setting up libjs-bootstrap (3.4.1+dfsg-1) ...\n",
            "Setting up python3-rdflib-jsonld (0.4.0-5) ...\n",
            "Setting up python3-sparqlwrapper (1.8.5-1) ...\n",
            "Setting up python3-schema-salad (5.0.20200220195218-1) ...\n",
            "Setting up nodejs (10.19.0~dfsg-3ubuntu1) ...\n",
            "update-alternatives: using /usr/bin/nodejs to provide /usr/bin/js (js) in auto mode\n",
            "Setting up python3-networkx (2.4-3ubuntu1) ...\n",
            "/usr/lib/python3/dist-packages/networkx/readwrite/tests/test_graphml.py:933: SyntaxWarning: \"is\" with a literal. Did you mean \"==\"?\n",
            "  assert H.nodes['n1']['special'] is 0\n",
            "/usr/lib/python3/dist-packages/networkx/readwrite/tests/test_graphml.py:935: SyntaxWarning: \"is\" with a literal. Did you mean \"==\"?\n",
            "  assert H.edges['n0','n1',1]['special'] is 0\n",
            "Setting up python3-bs4 (4.8.2-1) ...\n",
            "Setting up python3-prov (1.5.2-2) ...\n",
            "Setting up cwltool (2.0.20200224214940+dfsg-1) ...\n",
            "/usr/lib/python3/dist-packages/cwltool/docker.py:423: SyntaxWarning: \"is\" with a literal. Did you mean \"==\"?\n",
            "  if res_req is not None and (\"ramMin\" in res_req or \"ramMax\" is res_req):\n",
            "Processing triggers for fontconfig (2.13.1-2ubuntu3) ...\n",
            "Processing triggers for libc-bin (2.31-0ubuntu9.9) ...\n",
            "Processing triggers for man-db (2.9.1-1) ...\n"
          ]
        }
      ],
      "source": [
        "!apt-get install bowtie2 samtools"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 9,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "JgTq4lFjq93N",
        "outputId": "95f2b700-0c59-42a3-dea4-4ed2e0964055"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
            "Solving environment: / \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "  added / updated specs:\n",
            "    - bowtie2\n",
            "\n",
            "\n",
            "The following packages will be downloaded:\n",
            "\n",
            "    package                    |            build\n",
            "    ---------------------------|-----------------\n",
            "    bowtie2-2.5.1              |   py38h77f66f0_0        14.8 MB  bioconda\n",
            "    libhwloc-2.9.0             |       hd6dc26d_0         2.4 MB  conda-forge\n",
            "    perl-5.32.1                | 2_h7f98852_perl5        14.4 MB  conda-forge\n",
            "    tbb-2021.8.0               |       hf52228f_0         1.4 MB  conda-forge\n",
            "    ------------------------------------------------------------\n",
            "                                           Total:        33.1 MB\n",
            "\n",
            "The following NEW packages will be INSTALLED:\n",
            "\n",
            "  bowtie2            bioconda/linux-64::bowtie2-2.5.1-py38h77f66f0_0 \n",
            "  libhwloc           conda-forge/linux-64::libhwloc-2.9.0-hd6dc26d_0 \n",
            "  perl               conda-forge/linux-64::perl-5.32.1-2_h7f98852_perl5 \n",
            "  tbb                conda-forge/linux-64::tbb-2021.8.0-hf52228f_0 \n",
            "\n",
            "\n",
            "\n",
            "Downloading and Extracting Packages\n",
            "tbb-2021.8.0         | 1.4 MB    | :   0% 0/1 [00:00<?, ?it/s]\n",
            "perl-5.32.1          | 14.4 MB   | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\n",
            "\n",
            "bowtie2-2.5.1        | 14.8 MB   | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "tbb-2021.8.0         | 1.4 MB    | :   1% 0.010792658304727915/1 [00:00<00:26, 27.15s/it]\n",
            "\n",
            "\n",
            "libhwloc-2.9.0       | 2.4 MB    | :   1% 0.0063839528373015484/1 [00:00<00:42, 43.20s/it]\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "bowtie2-2.5.1        | 14.8 MB   | :   0% 0.0010587414082268809/1 [00:00<04:27, 267.59s/it]\u001b[A\u001b[A\n",
            "perl-5.32.1          | 14.4 MB   | :   0% 0.0010852662651499435/1 [00:00<04:41, 282.10s/it]\u001b[A\n",
            "\n",
            "bowtie2-2.5.1        | 14.8 MB   | :  17% 0.17363359094920847/1 [00:00<00:01,  1.73s/it]   \u001b[A\u001b[A\n",
            "perl-5.32.1          | 14.4 MB   | :  21% 0.21488272049968882/1 [00:00<00:01,  1.47s/it]   \u001b[A\n",
            "\n",
            "bowtie2-2.5.1        | 14.8 MB   | :  36% 0.36208956161359324/1 [00:00<00:00,  1.00s/it]\u001b[A\u001b[A\n",
            "tbb-2021.8.0         | 1.4 MB    | : 100% 1.0/1 [00:00<00:00,  2.30it/s]\n",
            "\n",
            "bowtie2-2.5.1        | 14.8 MB   | :  54% 0.542075601012163/1 [00:00<00:00,  1.25it/s]  \u001b[A\u001b[A\n",
            "perl-5.32.1          | 14.4 MB   | :  60% 0.5990669783627688/1 [00:00<00:00,  1.35it/s]\u001b[A\n",
            "\n",
            "bowtie2-2.5.1        | 14.8 MB   | :  75% 0.7538238826575392/1 [00:00<00:00,  1.50it/s]\u001b[A\u001b[A\n",
            "perl-5.32.1          | 14.4 MB   | :  79% 0.7944149060897586/1 [00:00<00:00,  1.53it/s]\u001b[A\n",
            "\n",
            "bowtie2-2.5.1        | 14.8 MB   | :  99% 0.9878057338756798/1 [00:00<00:00,  1.74it/s]\u001b[A\u001b[A\n",
            "perl-5.32.1          | 14.4 MB   | : 100% 1.0/1 [00:00<00:00,  1.69it/s]               \u001b[A\n",
            "\n",
            "\n",
            "libhwloc-2.9.0       | 2.4 MB    | : 100% 1.0/1 [00:00<00:00,  1.15it/s]                  \u001b[A\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "libhwloc-2.9.0       | 2.4 MB    | : 100% 1.0/1 [00:00<00:00,  1.15it/s]\u001b[A\u001b[A\u001b[A\n",
            "\n",
            "bowtie2-2.5.1        | 14.8 MB   | : 100% 1.0/1 [00:07<00:00,  1.74it/s]               \u001b[A\u001b[A\n",
            "                                                                        \n",
            "                                                                        \u001b[A\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\n",
            "Preparing transaction: | \b\b/ \b\bdone\n",
            "Verifying transaction: \\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n",
            "Executing transaction: - \b\b\\ \b\b| \b\b/ \b\bdone\n"
          ]
        }
      ],
      "source": [
        "!conda install -c bioconda bowtie2"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "vcr5CjibHTgF"
      },
      "source": [
        "## Установка FastQC"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 10,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "62l1Q2ymbENA",
        "outputId": "49e60c6f-90a3-4d47-d3da-c2df4029ab37"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "openjdk version \"17.0.3-internal\" 2022-04-19\n",
            "OpenJDK Runtime Environment (build 17.0.3-internal+0-adhoc..src)\n",
            "OpenJDK 64-Bit Server VM (build 17.0.3-internal+0-adhoc..src, mixed mode, sharing)\n"
          ]
        }
      ],
      "source": [
        "!java -version"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 11,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "8bpgwFgcbgRd",
        "outputId": "21dadef7-ab20-48ff-be87-1a3c2d92c015"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2023-03-06 15:58:31--  https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip\n",
            "Resolving www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)... 149.155.133.4\n",
            "Connecting to www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)|149.155.133.4|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 10249221 (9.8M) [application/zip]\n",
            "Saving to: ‘fastqc_v0.11.9.zip’\n",
            "\n",
            "fastqc_v0.11.9.zip  100%[===================>]   9.77M  1.53MB/s    in 6.4s    \n",
            "\n",
            "2023-03-06 15:58:42 (1.53 MB/s) - ‘fastqc_v0.11.9.zip’ saved [10249221/10249221]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!wget https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 12,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "yTcTjOTJblcm",
        "outputId": "9bb90f67-60dd-4ddc-f536-f8345f5c5ca7"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Archive:  fastqc_v0.11.9.zip\n",
            "  inflating: FastQC/cisd-jhdf5.jar   \n",
            "   creating: FastQC/Configuration/\n",
            "  inflating: FastQC/Configuration/adapter_list.txt  \n",
            "  inflating: FastQC/Configuration/contaminant_list.txt  \n",
            "  inflating: FastQC/Configuration/limits.txt  \n",
            "  inflating: FastQC/fastqc           \n",
            "  inflating: FastQC/fastqc_icon.ico  \n",
            "   creating: FastQC/Help/\n",
            "   creating: FastQC/Help/1 Introduction/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/\n",
            "  inflating: FastQC/Help/1 Introduction/.svn/entries  \n",
            "   creating: FastQC/Help/1 Introduction/.svn/props/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/text-base/\n",
            "  inflating: FastQC/Help/1 Introduction/.svn/text-base/1.1 What is FastQC.html.svn-base  \n",
            "   creating: FastQC/Help/1 Introduction/.svn/tmp/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/1 Introduction/1.1 What is FastQC.html  \n",
            "   creating: FastQC/Help/2 Basic Operations/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/\n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/entries  \n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/props/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/text-base/\n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.1 Opening a sequence file.html.svn-base  \n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.2 Evaluating Results.html.svn-base  \n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.3 Saving a Report.html.svn-base  \n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/tmp/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/2 Basic Operations/2.1 Opening a sequence file.html  \n",
            "  inflating: FastQC/Help/2 Basic Operations/2.2 Evaluating Results.html  \n",
            "  inflating: FastQC/Help/2 Basic Operations/2.3 Saving a Report.html  \n",
            "   creating: FastQC/Help/3 Analysis Modules/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/entries  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/prop-base/\n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/duplication_levels.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/prop-base/kmer_profiles.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_gc_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_n_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_sequence_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_sequence_gc_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_sequence_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_tile_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/sequence_length_distribution.png.svn-base  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/props/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/text-base/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/1 Basic Statistics.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/10 Adapter Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/11 Kmer Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/12 Per Tile Sequence Quality.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/2 Per Base Sequence Quality.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/3 Per Sequence Quality Scores.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/4 Per Base Sequence Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/5 Per Sequence GC Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/6 Per Base N Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/7 Sequence Length Distribution.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/8 Duplicate Sequences.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/9 Overrepresented Sequences.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/duplication_levels.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/kmer_profiles.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_gc_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_n_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_sequence_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_sequence_gc_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_sequence_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_tile_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/sequence_length_distribution.png.svn-base  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/tmp/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/1 Basic Statistics.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/10 Adapter Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/11 Kmer Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/12 Per Tile Sequence Quality.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/2 Per Base Sequence Quality.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/3 Per Sequence Quality Scores.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/4 Per Base Sequence Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/5 Per Sequence GC Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/6 Per Base N Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/7 Sequence Length Distribution.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/8 Duplicate Sequences.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/9 Overrepresented Sequences.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/duplication_levels.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/kmer_profiles.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_gc_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_n_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_sequence_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_sequence_gc_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_sequence_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_tile_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/sequence_length_distribution.png  \n",
            "  inflating: FastQC/INSTALL.txt      \n",
            "  inflating: FastQC/jbzip2-0.9.jar   \n",
            "  inflating: FastQC/LICENSE          \n",
            "  inflating: FastQC/LICENSE.txt      \n",
            "  inflating: FastQC/LICENSE_JHDF5.txt  \n",
            "   creating: FastQC/net/\n",
            "   creating: FastQC/net/sourceforge/\n",
            "   creating: FastQC/net/sourceforge/iharder/\n",
            "   creating: FastQC/net/sourceforge/iharder/base64/\n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$1.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$InputStream.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$OutputStream.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64.class  \n",
            "   creating: FastQC/org/\n",
            "   creating: FastQC/org/apache/\n",
            "   creating: FastQC/org/apache/commons/\n",
            "   creating: FastQC/org/apache/commons/math3/\n",
            "   creating: FastQC/org/apache/commons/math3/analysis/\n",
            "   creating: FastQC/org/apache/commons/math3/analysis/solvers/\n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/AbstractUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/AllowedSolution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BaseAbstractUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BaseUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BracketedUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BrentSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/UnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/UnivariateSolverUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/UnivariateFunction.class  \n",
            "   creating: FastQC/org/apache/commons/math3/distribution/\n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractIntegerDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractRealDistribution$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractRealDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/BetaDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/BinomialDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/CauchyDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/ChiSquaredDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/FDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/GammaDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/HypergeometricDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/IntegerDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/NormalDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/PascalDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/PoissonDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/RealDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/SaddlePointExpansion.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/TDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/WeibullDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/ZipfDistribution.class  \n",
            "   creating: FastQC/org/apache/commons/math3/exception/\n",
            "  inflating: FastQC/org/apache/commons/math3/exception/ConvergenceException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/DimensionMismatchException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathArithmeticException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalArgumentException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalNumberException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalStateException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathInternalError.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MaxCountExceededException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NoBracketingException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotFiniteNumberException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotPositiveException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotStrictlyPositiveException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NullArgumentException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NumberIsTooLargeException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NumberIsTooSmallException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/OutOfRangeException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/TooManyEvaluationsException.class  \n",
            "   creating: FastQC/org/apache/commons/math3/exception/util/\n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ArgUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ExceptionContext.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ExceptionContextProvider.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/Localizable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/LocalizedFormats.class  \n",
            "   creating: FastQC/org/apache/commons/math3/random/\n",
            "  inflating: FastQC/org/apache/commons/math3/random/AbstractWell.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/BitsStreamGenerator.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomData.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomDataImpl.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomGenerator.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/Well19937c.class  \n",
            "   creating: FastQC/org/apache/commons/math3/special/\n",
            "  inflating: FastQC/org/apache/commons/math3/special/Beta$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Beta.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Erf.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Gamma$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Gamma.class  \n",
            "   creating: FastQC/org/apache/commons/math3/util/\n",
            "  inflating: FastQC/org/apache/commons/math3/util/ArithmeticUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/ContinuedFraction.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/DoubleArray.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$ExpFracTable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$ExpIntTable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$lnMant.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMathCalc.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMathLiteralArrays.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor$MaxCountExceededCallback.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/MathUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Precision.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/ResizableDoubleArray.class  \n",
            "  inflating: FastQC/README.md        \n",
            "  inflating: FastQC/README.txt       \n",
            "  inflating: FastQC/RELEASE_NOTES.txt  \n",
            "  inflating: FastQC/run_fastqc.bat   \n",
            "  inflating: FastQC/sam-1.103.jar    \n",
            "   creating: FastQC/Templates/\n",
            "  inflating: FastQC/Templates/fastqc2fo.xsl  \n",
            "  inflating: FastQC/Templates/header_template.html  \n",
            "   creating: FastQC/Templates/Icons/\n",
            " extracting: FastQC/Templates/Icons/error.png  \n",
            " extracting: FastQC/Templates/Icons/fastqc_icon.png  \n",
            " extracting: FastQC/Templates/Icons/tick.png  \n",
            " extracting: FastQC/Templates/Icons/warning.png  \n",
            "   creating: FastQC/uk/\n",
            "   creating: FastQC/uk/ac/\n",
            "   creating: FastQC/uk/ac/babraham/\n",
            "   creating: FastQC/uk/ac/babraham/FastQC/\n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Analysis/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisListener.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisQueue.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisRunner.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/OfflineRunner.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Dialogs/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/AboutDialog$1.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/AboutDialog.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/FastQCTitlePanel$SmoothJLabel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/FastQCTitlePanel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/WelcomePanel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCApplication$1.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCApplication.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCConfig.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCMenuBar.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/FileFilters/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/BAMFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/CasavaFastQFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/FastQFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/GobyFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/MappedBAMFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/SequenceFileFilter.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Graphs/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/BaseGroup.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/LineGraph.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/QualityBoxPlot.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/TileGraph.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Help/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpDialog.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpIndexRoot$FileSorter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpIndexRoot.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPage.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPageDisplay$HelpEditor.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPageDisplay.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpSearchPanel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Modules/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AbstractQCModule.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent$Adapter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/BasicStats$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/BasicStats.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/DuplicationLevel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/GCModel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/GCModelValue.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent$Kmer.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/ModuleConfig.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/ModuleFactory.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/NContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs$OverrepresentedSeq.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerBaseQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerBaseSequenceContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerSequenceGCContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerSequenceQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerTileQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/QCModule.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/SequenceLengthDistribution.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Report/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Report/HTMLReportArchive.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Report/stylesheet.txt  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Resources/\n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/error.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon.png  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon.svg  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon_100.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/tick.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/warning.png  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Results/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Results/ResultsPanel$ModuleRenderer.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Results/ResultsPanel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/BAMFile.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/Contaminant.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/ContaminantHit.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/ContaminentFinder.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Fast5File.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/FastQFile.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/QualityEncoding/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/QualityEncoding/PhredEncoding.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Sequence.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFactory.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFile.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFileGroup.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFormatException.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Statistics/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Statistics/NormalDistribution.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Statistics/PearsonCorrelation.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Utilities/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/CasavaBasename.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/HotColdColourGradient.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/ImageToBase64.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/MultiMemberGZIPInputStream.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/NameFormatException.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/NanoporeBasename.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/QualityCount.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/RGB.class  \n"
          ]
        }
      ],
      "source": [
        "!unzip fastqc_v0.11.9.zip"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 13,
      "metadata": {
        "id": "S79LH0aXbvLx"
      },
      "outputs": [],
      "source": [
        "!chmod a+x FastQC/fastqc"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 14,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "PsLuZDt4b7jT",
        "outputId": "eacde585-672d-495d-8505-1a1c57c934b6"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "            FastQC - A high throughput sequence QC analysis tool\n",
            "\n",
            "SYNOPSIS\n",
            "\n",
            "\tfastqc seqfile1 seqfile2 .. seqfileN\n",
            "\n",
            "    fastqc [-o output dir] [--(no)extract] [-f fastq|bam|sam] \n",
            "           [-c contaminant file] seqfile1 .. seqfileN\n",
            "\n",
            "DESCRIPTION\n",
            "\n",
            "    FastQC reads a set of sequence files and produces from each one a quality\n",
            "    control report consisting of a number of different modules, each one of \n",
            "    which will help to identify a different potential type of problem in your\n",
            "    data.\n",
            "    \n",
            "    If no files to process are specified on the command line then the program\n",
            "    will start as an interactive graphical application.  If files are provided\n",
            "    on the command line then the program will run with no user interaction\n",
            "    required.  In this mode it is suitable for inclusion into a standardised\n",
            "    analysis pipeline.\n",
            "    \n",
            "    The options for the program as as follows:\n",
            "    \n",
            "    -h --help       Print this help file and exit\n",
            "    \n",
            "    -v --version    Print the version of the program and exit\n",
            "    \n",
            "    -o --outdir     Create all output files in the specified output directory.\n",
            "                    Please note that this directory must exist as the program\n",
            "                    will not create it.  If this option is not set then the \n",
            "                    output file for each sequence file is created in the same\n",
            "                    directory as the sequence file which was processed.\n",
            "                    \n",
            "    --casava        Files come from raw casava output. Files in the same sample\n",
            "                    group (differing only by the group number) will be analysed\n",
            "                    as a set rather than individually. Sequences with the filter\n",
            "                    flag set in the header will be excluded from the analysis.\n",
            "                    Files must have the same names given to them by casava\n",
            "                    (including being gzipped and ending with .gz) otherwise they\n",
            "                    won't be grouped together correctly.\n",
            "                    \n",
            "    --nano          Files come from nanopore sequences and are in fast5 format. In\n",
            "                    this mode you can pass in directories to process and the program\n",
            "                    will take in all fast5 files within those directories and produce\n",
            "                    a single output file from the sequences found in all files.                    \n",
            "                    \n",
            "    --nofilter      If running with --casava then don't remove read flagged by\n",
            "                    casava as poor quality when performing the QC analysis.\n",
            "                   \n",
            "    --extract       If set then the zipped output file will be uncompressed in\n",
            "                    the same directory after it has been created.  By default\n",
            "                    this option will be set if fastqc is run in non-interactive\n",
            "                    mode.\n",
            "                    \n",
            "    -j --java       Provides the full path to the java binary you want to use to\n",
            "                    launch fastqc. If not supplied then java is assumed to be in\n",
            "                    your path.\n",
            "                   \n",
            "    --noextract     Do not uncompress the output file after creating it.  You\n",
            "                    should set this option if you do not wish to uncompress\n",
            "                    the output when running in non-interactive mode.\n",
            "                    \n",
            "    --nogroup       Disable grouping of bases for reads >50bp. All reports will\n",
            "                    show data for every base in the read.  WARNING: Using this\n",
            "                    option will cause fastqc to crash and burn if you use it on\n",
            "                    really long reads, and your plots may end up a ridiculous size.\n",
            "                    You have been warned!\n",
            "                    \n",
            "    --min_length    Sets an artificial lower limit on the length of the sequence\n",
            "                    to be shown in the report.  As long as you set this to a value\n",
            "                    greater or equal to your longest read length then this will be\n",
            "                    the sequence length used to create your read groups.  This can\n",
            "                    be useful for making directly comaparable statistics from \n",
            "                    datasets with somewhat variable read lengths.\n",
            "                    \n",
            "    -f --format     Bypasses the normal sequence file format detection and\n",
            "                    forces the program to use the specified format.  Valid\n",
            "                    formats are bam,sam,bam_mapped,sam_mapped and fastq\n",
            "                    \n",
            "    -t --threads    Specifies the number of files which can be processed\n",
            "                    simultaneously.  Each thread will be allocated 250MB of\n",
            "                    memory so you shouldn't run more threads than your\n",
            "                    available memory will cope with, and not more than\n",
            "                    6 threads on a 32 bit machine\n",
            "                  \n",
            "    -c              Specifies a non-default file which contains the list of\n",
            "    --contaminants  contaminants to screen overrepresented sequences against.\n",
            "                    The file must contain sets of named contaminants in the\n",
            "                    form name[tab]sequence.  Lines prefixed with a hash will\n",
            "                    be ignored.\n",
            "\n",
            "    -a              Specifies a non-default file which contains the list of\n",
            "    --adapters      adapter sequences which will be explicity searched against\n",
            "                    the library. The file must contain sets of named adapters\n",
            "                    in the form name[tab]sequence.  Lines prefixed with a hash\n",
            "                    will be ignored.\n",
            "                    \n",
            "    -l              Specifies a non-default file which contains a set of criteria\n",
            "    --limits        which will be used to determine the warn/error limits for the\n",
            "                    various modules.  This file can also be used to selectively \n",
            "                    remove some modules from the output all together.  The format\n",
            "                    needs to mirror the default limits.txt file found in the\n",
            "                    Configuration folder.\n",
            "                    \n",
            "   -k --kmers       Specifies the length of Kmer to look for in the Kmer content\n",
            "                    module. Specified Kmer length must be between 2 and 10. Default\n",
            "                    length is 7 if not specified.\n",
            "                    \n",
            "   -q --quiet       Supress all progress messages on stdout and only report errors.\n",
            "   \n",
            "   -d --dir         Selects a directory to be used for temporary files written when\n",
            "                    generating report images. Defaults to system temp directory if\n",
            "                    not specified.\n",
            "                    \n",
            "BUGS\n",
            "\n",
            "    Any bugs in fastqc should be reported either to simon.andrews@babraham.ac.uk\n",
            "    or in www.bioinformatics.babraham.ac.uk/bugzilla/\n",
            "                   \n",
            "    \n"
          ]
        }
      ],
      "source": [
        "!./FastQC/fastqc --help"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "10Fs3qrmHmJi"
      },
      "source": [
        "# Выравнивание"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "-z4GQiMfU1wf"
      },
      "source": [
        "## Скачиваем чтения\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 15,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "lFWnMg0uU0sJ",
        "outputId": "9f59ca81-7031-4762-e606-696b42ccdd7c"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2023-03-06 15:58:42--  https://www.encodeproject.org/files/ENCFF097DJJ/@@download/ENCFF097DJJ.fastq.gz\n",
            "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
            "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
            "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
            "Location: https://encode-public.s3.amazonaws.com/2022/01/27/55971b7e-a7ed-4b41-a1cc-ac77d6d28581/ENCFF097DJJ.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF097DJJ.fastq.gz&AWSAccessKeyId=ASIATGZNGCNX6XA2USS7&Signature=wYog0DljHqdbQfyXGWFahOEWNkQ%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEF8aCXVzLXdlc3QtMiJHMEUCIEf0njY0gn5qPzllWz2%2BCAA5GQYx%2FEWbELmBaB6Rjud%2BAiEAv98zukUah4nIkPJm8RBZixDu7d5aeqwzdFAr6oaiBFkqsgUIGBAAGgwyMjA3NDg3MTQ4NjMiDMaIbwbF67CBfGSnuCqPBUTGrA4qk5RWDU43bvH1IAkM4h5nlru2SjcYEzs3ys4HaqqdC8EWHWZtgQkOcOJHbFqC6soIdCLUq49LhiwhCampX9JWcXdndP%2BEm5eFIcfQ9pHWLmh6zFy6V2mtpmgRyy3ebLdkrcZSmD4HzGB7fetSd4lZwa64vKcqv1Ei2odnE2%2Be%2B2emEqkhFbhofB%2Fn7BIx6kkdYcAufW1xP%2BvUUZ4EBhlysI0s20Bf9%2BqMolQryUHg0GPO9NVkupYV%2FbR8nKn26Tx%2FIqyGL4f53ZmyZsjmzAur5tSQnFlyaZGA2S5GYLvWpzWeIP9wvzvtbucV0xR72JyJiTaSv2PYV0ZOrS0zAoYC2XHzUTdLsJ%2BqDdySl4TZBCmXuqOOpq3JMHPY9SE6PeBsMXsqwChkEg63zG%2FDaIBJIvhotanG%2BZVztMkMSwiolR4jpKwQDCeo6WnzwZUdAw5loMHgPmuvZRLHhMR921Aq4WFrFG8ywUOSv3R%2B9nvnNvaZjuiZrGRNZma6zK7QkLwjo7pvXq6AJryXAGJD%2F1cnRNnB3bNKsGEaImSXd6mz1wrI8KNMTMUl15Umdd7uR9c%2FIX9GYmkF2JLA8ElRgGZgrCIPTrEI3z5Wu%2BB2kUpxhbTLrI0AjegwlIN3%2F8NvoLMO9FL1hJaIAHeLaXjDtA5i596gxhg3SNfpL25oB%2Fj%2Fo8AfjJpYjKt%2Ff85ZtjIOiyJwFZwD1U1Z%2Fm9LWcqz1wd6mslSBZ%2BxMarnz3taS2CHclZhrEbohAqsCz%2BNG98kpG3iB3IAacgSMGNMNAPCdFV8dErBr99OuX7fguE8svrfFAbOnsx8PPXxAnQ4qVmTLdRlDiI4Zw%2Bvolgrni5Jhc6lutX5l2bvDZ5pPNEwvIWYoAY6sQEW2CkTtPVG8UN2Gr0CVUSXOTMy89oxWrWdBvM6Ex0xffXnb65Pd3q89GmQmBgNvkExkHfBtxdfgYumYiqyggUDg1ZJI%2BTTPwTWAu1ctf8zNjohDWgf%2B1X21rWKqWEQtBl4qOjaSkApU8vnJDI%2BHdPtxxHie3gKaaSnMgLKhgp3IfnOA0JLl6MSgVpa0s%2FncKSAa4SxOutOKtwg8SwKQ2E2cQBiSp85fVnFLKZlrndgD0M%3D&Expires=1678247923 [following]\n",
            "--2023-03-06 15:58:43--  https://encode-public.s3.amazonaws.com/2022/01/27/55971b7e-a7ed-4b41-a1cc-ac77d6d28581/ENCFF097DJJ.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF097DJJ.fastq.gz&AWSAccessKeyId=ASIATGZNGCNX6XA2USS7&Signature=wYog0DljHqdbQfyXGWFahOEWNkQ%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEF8aCXVzLXdlc3QtMiJHMEUCIEf0njY0gn5qPzllWz2%2BCAA5GQYx%2FEWbELmBaB6Rjud%2BAiEAv98zukUah4nIkPJm8RBZixDu7d5aeqwzdFAr6oaiBFkqsgUIGBAAGgwyMjA3NDg3MTQ4NjMiDMaIbwbF67CBfGSnuCqPBUTGrA4qk5RWDU43bvH1IAkM4h5nlru2SjcYEzs3ys4HaqqdC8EWHWZtgQkOcOJHbFqC6soIdCLUq49LhiwhCampX9JWcXdndP%2BEm5eFIcfQ9pHWLmh6zFy6V2mtpmgRyy3ebLdkrcZSmD4HzGB7fetSd4lZwa64vKcqv1Ei2odnE2%2Be%2B2emEqkhFbhofB%2Fn7BIx6kkdYcAufW1xP%2BvUUZ4EBhlysI0s20Bf9%2BqMolQryUHg0GPO9NVkupYV%2FbR8nKn26Tx%2FIqyGL4f53ZmyZsjmzAur5tSQnFlyaZGA2S5GYLvWpzWeIP9wvzvtbucV0xR72JyJiTaSv2PYV0ZOrS0zAoYC2XHzUTdLsJ%2BqDdySl4TZBCmXuqOOpq3JMHPY9SE6PeBsMXsqwChkEg63zG%2FDaIBJIvhotanG%2BZVztMkMSwiolR4jpKwQDCeo6WnzwZUdAw5loMHgPmuvZRLHhMR921Aq4WFrFG8ywUOSv3R%2B9nvnNvaZjuiZrGRNZma6zK7QkLwjo7pvXq6AJryXAGJD%2F1cnRNnB3bNKsGEaImSXd6mz1wrI8KNMTMUl15Umdd7uR9c%2FIX9GYmkF2JLA8ElRgGZgrCIPTrEI3z5Wu%2BB2kUpxhbTLrI0AjegwlIN3%2F8NvoLMO9FL1hJaIAHeLaXjDtA5i596gxhg3SNfpL25oB%2Fj%2Fo8AfjJpYjKt%2Ff85ZtjIOiyJwFZwD1U1Z%2Fm9LWcqz1wd6mslSBZ%2BxMarnz3taS2CHclZhrEbohAqsCz%2BNG98kpG3iB3IAacgSMGNMNAPCdFV8dErBr99OuX7fguE8svrfFAbOnsx8PPXxAnQ4qVmTLdRlDiI4Zw%2Bvolgrni5Jhc6lutX5l2bvDZ5pPNEwvIWYoAY6sQEW2CkTtPVG8UN2Gr0CVUSXOTMy89oxWrWdBvM6Ex0xffXnb65Pd3q89GmQmBgNvkExkHfBtxdfgYumYiqyggUDg1ZJI%2BTTPwTWAu1ctf8zNjohDWgf%2B1X21rWKqWEQtBl4qOjaSkApU8vnJDI%2BHdPtxxHie3gKaaSnMgLKhgp3IfnOA0JLl6MSgVpa0s%2FncKSAa4SxOutOKtwg8SwKQ2E2cQBiSp85fVnFLKZlrndgD0M%3D&Expires=1678247923\n",
            "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.92.129.217, 52.92.250.177, 52.92.160.89, ...\n",
            "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.92.129.217|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 1492781260 (1.4G) [binary/octet-stream]\n",
            "Saving to: ‘ENCFF097DJJ.fastq.gz’\n",
            "\n",
            "ENCFF097DJJ.fastq.g 100%[===================>]   1.39G  21.2MB/s    in 69s     \n",
            "\n",
            "2023-03-06 15:59:52 (20.8 MB/s) - ‘ENCFF097DJJ.fastq.gz’ saved [1492781260/1492781260]\n",
            "\n",
            "--2023-03-06 15:59:52--  https://www.encodeproject.org/files/ENCFF161DOP/@@download/ENCFF161DOP.fastq.gz\n",
            "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
            "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
            "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
            "Location: https://encode-public.s3.amazonaws.com/2022/01/27/f0523bf4-88c8-4696-881c-dce530b7b3a5/ENCFF161DOP.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF161DOP.fastq.gz&AWSAccessKeyId=ASIATGZNGCNX6XA2USS7&Signature=QbAlhUg%2BmRM1pZ3Lb3QfnT7p3Po%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEF8aCXVzLXdlc3QtMiJHMEUCIEf0njY0gn5qPzllWz2%2BCAA5GQYx%2FEWbELmBaB6Rjud%2BAiEAv98zukUah4nIkPJm8RBZixDu7d5aeqwzdFAr6oaiBFkqsgUIGBAAGgwyMjA3NDg3MTQ4NjMiDMaIbwbF67CBfGSnuCqPBUTGrA4qk5RWDU43bvH1IAkM4h5nlru2SjcYEzs3ys4HaqqdC8EWHWZtgQkOcOJHbFqC6soIdCLUq49LhiwhCampX9JWcXdndP%2BEm5eFIcfQ9pHWLmh6zFy6V2mtpmgRyy3ebLdkrcZSmD4HzGB7fetSd4lZwa64vKcqv1Ei2odnE2%2Be%2B2emEqkhFbhofB%2Fn7BIx6kkdYcAufW1xP%2BvUUZ4EBhlysI0s20Bf9%2BqMolQryUHg0GPO9NVkupYV%2FbR8nKn26Tx%2FIqyGL4f53ZmyZsjmzAur5tSQnFlyaZGA2S5GYLvWpzWeIP9wvzvtbucV0xR72JyJiTaSv2PYV0ZOrS0zAoYC2XHzUTdLsJ%2BqDdySl4TZBCmXuqOOpq3JMHPY9SE6PeBsMXsqwChkEg63zG%2FDaIBJIvhotanG%2BZVztMkMSwiolR4jpKwQDCeo6WnzwZUdAw5loMHgPmuvZRLHhMR921Aq4WFrFG8ywUOSv3R%2B9nvnNvaZjuiZrGRNZma6zK7QkLwjo7pvXq6AJryXAGJD%2F1cnRNnB3bNKsGEaImSXd6mz1wrI8KNMTMUl15Umdd7uR9c%2FIX9GYmkF2JLA8ElRgGZgrCIPTrEI3z5Wu%2BB2kUpxhbTLrI0AjegwlIN3%2F8NvoLMO9FL1hJaIAHeLaXjDtA5i596gxhg3SNfpL25oB%2Fj%2Fo8AfjJpYjKt%2Ff85ZtjIOiyJwFZwD1U1Z%2Fm9LWcqz1wd6mslSBZ%2BxMarnz3taS2CHclZhrEbohAqsCz%2BNG98kpG3iB3IAacgSMGNMNAPCdFV8dErBr99OuX7fguE8svrfFAbOnsx8PPXxAnQ4qVmTLdRlDiI4Zw%2Bvolgrni5Jhc6lutX5l2bvDZ5pPNEwvIWYoAY6sQEW2CkTtPVG8UN2Gr0CVUSXOTMy89oxWrWdBvM6Ex0xffXnb65Pd3q89GmQmBgNvkExkHfBtxdfgYumYiqyggUDg1ZJI%2BTTPwTWAu1ctf8zNjohDWgf%2B1X21rWKqWEQtBl4qOjaSkApU8vnJDI%2BHdPtxxHie3gKaaSnMgLKhgp3IfnOA0JLl6MSgVpa0s%2FncKSAa4SxOutOKtwg8SwKQ2E2cQBiSp85fVnFLKZlrndgD0M%3D&Expires=1678247993 [following]\n",
            "--2023-03-06 15:59:53--  https://encode-public.s3.amazonaws.com/2022/01/27/f0523bf4-88c8-4696-881c-dce530b7b3a5/ENCFF161DOP.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF161DOP.fastq.gz&AWSAccessKeyId=ASIATGZNGCNX6XA2USS7&Signature=QbAlhUg%2BmRM1pZ3Lb3QfnT7p3Po%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEF8aCXVzLXdlc3QtMiJHMEUCIEf0njY0gn5qPzllWz2%2BCAA5GQYx%2FEWbELmBaB6Rjud%2BAiEAv98zukUah4nIkPJm8RBZixDu7d5aeqwzdFAr6oaiBFkqsgUIGBAAGgwyMjA3NDg3MTQ4NjMiDMaIbwbF67CBfGSnuCqPBUTGrA4qk5RWDU43bvH1IAkM4h5nlru2SjcYEzs3ys4HaqqdC8EWHWZtgQkOcOJHbFqC6soIdCLUq49LhiwhCampX9JWcXdndP%2BEm5eFIcfQ9pHWLmh6zFy6V2mtpmgRyy3ebLdkrcZSmD4HzGB7fetSd4lZwa64vKcqv1Ei2odnE2%2Be%2B2emEqkhFbhofB%2Fn7BIx6kkdYcAufW1xP%2BvUUZ4EBhlysI0s20Bf9%2BqMolQryUHg0GPO9NVkupYV%2FbR8nKn26Tx%2FIqyGL4f53ZmyZsjmzAur5tSQnFlyaZGA2S5GYLvWpzWeIP9wvzvtbucV0xR72JyJiTaSv2PYV0ZOrS0zAoYC2XHzUTdLsJ%2BqDdySl4TZBCmXuqOOpq3JMHPY9SE6PeBsMXsqwChkEg63zG%2FDaIBJIvhotanG%2BZVztMkMSwiolR4jpKwQDCeo6WnzwZUdAw5loMHgPmuvZRLHhMR921Aq4WFrFG8ywUOSv3R%2B9nvnNvaZjuiZrGRNZma6zK7QkLwjo7pvXq6AJryXAGJD%2F1cnRNnB3bNKsGEaImSXd6mz1wrI8KNMTMUl15Umdd7uR9c%2FIX9GYmkF2JLA8ElRgGZgrCIPTrEI3z5Wu%2BB2kUpxhbTLrI0AjegwlIN3%2F8NvoLMO9FL1hJaIAHeLaXjDtA5i596gxhg3SNfpL25oB%2Fj%2Fo8AfjJpYjKt%2Ff85ZtjIOiyJwFZwD1U1Z%2Fm9LWcqz1wd6mslSBZ%2BxMarnz3taS2CHclZhrEbohAqsCz%2BNG98kpG3iB3IAacgSMGNMNAPCdFV8dErBr99OuX7fguE8svrfFAbOnsx8PPXxAnQ4qVmTLdRlDiI4Zw%2Bvolgrni5Jhc6lutX5l2bvDZ5pPNEwvIWYoAY6sQEW2CkTtPVG8UN2Gr0CVUSXOTMy89oxWrWdBvM6Ex0xffXnb65Pd3q89GmQmBgNvkExkHfBtxdfgYumYiqyggUDg1ZJI%2BTTPwTWAu1ctf8zNjohDWgf%2B1X21rWKqWEQtBl4qOjaSkApU8vnJDI%2BHdPtxxHie3gKaaSnMgLKhgp3IfnOA0JLl6MSgVpa0s%2FncKSAa4SxOutOKtwg8SwKQ2E2cQBiSp85fVnFLKZlrndgD0M%3D&Expires=1678247993\n",
            "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.218.243.161, 52.92.196.49, 52.218.184.51, ...\n",
            "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.218.243.161|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 2998652247 (2.8G) [binary/octet-stream]\n",
            "Saving to: ‘ENCFF161DOP.fastq.gz’\n",
            "\n",
            "ENCFF161DOP.fastq.g 100%[===================>]   2.79G  20.1MB/s    in 2m 20s  \n",
            "\n",
            "2023-03-06 16:02:14 (20.4 MB/s) - ‘ENCFF161DOP.fastq.gz’ saved [2998652247/2998652247]\n",
            "\n",
            "--2023-03-06 16:02:14--  https://www.encodeproject.org/files/ENCFF259JCU/@@download/ENCFF259JCU.fastq.gz\n",
            "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
            "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
            "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
            "Location: https://encode-public.s3.amazonaws.com/2022/01/27/0f5cce95-0d18-413b-853b-00102fc120d8/ENCFF259JCU.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF259JCU.fastq.gz&AWSAccessKeyId=ASIATGZNGCNX6XA2USS7&Signature=WF0XvgKsxMZ5uvhEVHYzk116cMw%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEF8aCXVzLXdlc3QtMiJHMEUCIEf0njY0gn5qPzllWz2%2BCAA5GQYx%2FEWbELmBaB6Rjud%2BAiEAv98zukUah4nIkPJm8RBZixDu7d5aeqwzdFAr6oaiBFkqsgUIGBAAGgwyMjA3NDg3MTQ4NjMiDMaIbwbF67CBfGSnuCqPBUTGrA4qk5RWDU43bvH1IAkM4h5nlru2SjcYEzs3ys4HaqqdC8EWHWZtgQkOcOJHbFqC6soIdCLUq49LhiwhCampX9JWcXdndP%2BEm5eFIcfQ9pHWLmh6zFy6V2mtpmgRyy3ebLdkrcZSmD4HzGB7fetSd4lZwa64vKcqv1Ei2odnE2%2Be%2B2emEqkhFbhofB%2Fn7BIx6kkdYcAufW1xP%2BvUUZ4EBhlysI0s20Bf9%2BqMolQryUHg0GPO9NVkupYV%2FbR8nKn26Tx%2FIqyGL4f53ZmyZsjmzAur5tSQnFlyaZGA2S5GYLvWpzWeIP9wvzvtbucV0xR72JyJiTaSv2PYV0ZOrS0zAoYC2XHzUTdLsJ%2BqDdySl4TZBCmXuqOOpq3JMHPY9SE6PeBsMXsqwChkEg63zG%2FDaIBJIvhotanG%2BZVztMkMSwiolR4jpKwQDCeo6WnzwZUdAw5loMHgPmuvZRLHhMR921Aq4WFrFG8ywUOSv3R%2B9nvnNvaZjuiZrGRNZma6zK7QkLwjo7pvXq6AJryXAGJD%2F1cnRNnB3bNKsGEaImSXd6mz1wrI8KNMTMUl15Umdd7uR9c%2FIX9GYmkF2JLA8ElRgGZgrCIPTrEI3z5Wu%2BB2kUpxhbTLrI0AjegwlIN3%2F8NvoLMO9FL1hJaIAHeLaXjDtA5i596gxhg3SNfpL25oB%2Fj%2Fo8AfjJpYjKt%2Ff85ZtjIOiyJwFZwD1U1Z%2Fm9LWcqz1wd6mslSBZ%2BxMarnz3taS2CHclZhrEbohAqsCz%2BNG98kpG3iB3IAacgSMGNMNAPCdFV8dErBr99OuX7fguE8svrfFAbOnsx8PPXxAnQ4qVmTLdRlDiI4Zw%2Bvolgrni5Jhc6lutX5l2bvDZ5pPNEwvIWYoAY6sQEW2CkTtPVG8UN2Gr0CVUSXOTMy89oxWrWdBvM6Ex0xffXnb65Pd3q89GmQmBgNvkExkHfBtxdfgYumYiqyggUDg1ZJI%2BTTPwTWAu1ctf8zNjohDWgf%2B1X21rWKqWEQtBl4qOjaSkApU8vnJDI%2BHdPtxxHie3gKaaSnMgLKhgp3IfnOA0JLl6MSgVpa0s%2FncKSAa4SxOutOKtwg8SwKQ2E2cQBiSp85fVnFLKZlrndgD0M%3D&Expires=1678248134 [following]\n",
            "--2023-03-06 16:02:14--  https://encode-public.s3.amazonaws.com/2022/01/27/0f5cce95-0d18-413b-853b-00102fc120d8/ENCFF259JCU.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF259JCU.fastq.gz&AWSAccessKeyId=ASIATGZNGCNX6XA2USS7&Signature=WF0XvgKsxMZ5uvhEVHYzk116cMw%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEF8aCXVzLXdlc3QtMiJHMEUCIEf0njY0gn5qPzllWz2%2BCAA5GQYx%2FEWbELmBaB6Rjud%2BAiEAv98zukUah4nIkPJm8RBZixDu7d5aeqwzdFAr6oaiBFkqsgUIGBAAGgwyMjA3NDg3MTQ4NjMiDMaIbwbF67CBfGSnuCqPBUTGrA4qk5RWDU43bvH1IAkM4h5nlru2SjcYEzs3ys4HaqqdC8EWHWZtgQkOcOJHbFqC6soIdCLUq49LhiwhCampX9JWcXdndP%2BEm5eFIcfQ9pHWLmh6zFy6V2mtpmgRyy3ebLdkrcZSmD4HzGB7fetSd4lZwa64vKcqv1Ei2odnE2%2Be%2B2emEqkhFbhofB%2Fn7BIx6kkdYcAufW1xP%2BvUUZ4EBhlysI0s20Bf9%2BqMolQryUHg0GPO9NVkupYV%2FbR8nKn26Tx%2FIqyGL4f53ZmyZsjmzAur5tSQnFlyaZGA2S5GYLvWpzWeIP9wvzvtbucV0xR72JyJiTaSv2PYV0ZOrS0zAoYC2XHzUTdLsJ%2BqDdySl4TZBCmXuqOOpq3JMHPY9SE6PeBsMXsqwChkEg63zG%2FDaIBJIvhotanG%2BZVztMkMSwiolR4jpKwQDCeo6WnzwZUdAw5loMHgPmuvZRLHhMR921Aq4WFrFG8ywUOSv3R%2B9nvnNvaZjuiZrGRNZma6zK7QkLwjo7pvXq6AJryXAGJD%2F1cnRNnB3bNKsGEaImSXd6mz1wrI8KNMTMUl15Umdd7uR9c%2FIX9GYmkF2JLA8ElRgGZgrCIPTrEI3z5Wu%2BB2kUpxhbTLrI0AjegwlIN3%2F8NvoLMO9FL1hJaIAHeLaXjDtA5i596gxhg3SNfpL25oB%2Fj%2Fo8AfjJpYjKt%2Ff85ZtjIOiyJwFZwD1U1Z%2Fm9LWcqz1wd6mslSBZ%2BxMarnz3taS2CHclZhrEbohAqsCz%2BNG98kpG3iB3IAacgSMGNMNAPCdFV8dErBr99OuX7fguE8svrfFAbOnsx8PPXxAnQ4qVmTLdRlDiI4Zw%2Bvolgrni5Jhc6lutX5l2bvDZ5pPNEwvIWYoAY6sQEW2CkTtPVG8UN2Gr0CVUSXOTMy89oxWrWdBvM6Ex0xffXnb65Pd3q89GmQmBgNvkExkHfBtxdfgYumYiqyggUDg1ZJI%2BTTPwTWAu1ctf8zNjohDWgf%2B1X21rWKqWEQtBl4qOjaSkApU8vnJDI%2BHdPtxxHie3gKaaSnMgLKhgp3IfnOA0JLl6MSgVpa0s%2FncKSAa4SxOutOKtwg8SwKQ2E2cQBiSp85fVnFLKZlrndgD0M%3D&Expires=1678248134\n",
            "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.92.181.33, 52.218.182.11, 52.92.240.97, ...\n",
            "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.92.181.33|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 4506049076 (4.2G) [binary/octet-stream]\n",
            "Saving to: ‘ENCFF259JCU.fastq.gz’\n",
            "\n",
            "ENCFF259JCU.fastq.g 100%[===================>]   4.20G  22.0MB/s    in 3m 45s  \n",
            "\n",
            "2023-03-06 16:06:00 (19.1 MB/s) - ‘ENCFF259JCU.fastq.gz’ saved [4506049076/4506049076]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "# ChIP-seq на гистоновой метке\n",
        "!wget https://www.encodeproject.org/files/ENCFF097DJJ/@@download/ENCFF097DJJ.fastq.gz\n",
        "!wget https://www.encodeproject.org/files/ENCFF161DOP/@@download/ENCFF161DOP.fastq.gz\n",
        "# ChIP-seq контроль\n",
        "!wget https://www.encodeproject.org/files/ENCFF259JCU/@@download/ENCFF259JCU.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 16,
      "metadata": {
        "id": "v73fH7WiVDu9"
      },
      "outputs": [],
      "source": [
        "# распаковка ахрива\n",
        "!gzip -d /content/ENCFF097DJJ.fastq.gz /content/ENCFF161DOP.fastq.gz /content/ENCFF259JCU.fastq.gz"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "6sPcIefNHx-L"
      },
      "source": [
        "## FastQC"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 17,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "5XM6jBcuGjMV",
        "outputId": "0a5d1856-5f0c-410a-ee9f-3206ee311e45"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Started analysis of ENCFF097DJJ.fastq\n",
            "Approx 5% complete for ENCFF097DJJ.fastq\n",
            "Approx 10% complete for ENCFF097DJJ.fastq\n",
            "Approx 15% complete for ENCFF097DJJ.fastq\n",
            "Approx 20% complete for ENCFF097DJJ.fastq\n",
            "Approx 25% complete for ENCFF097DJJ.fastq\n",
            "Approx 30% complete for ENCFF097DJJ.fastq\n",
            "Approx 35% complete for ENCFF097DJJ.fastq\n",
            "Approx 40% complete for ENCFF097DJJ.fastq\n",
            "Approx 45% complete for ENCFF097DJJ.fastq\n",
            "Approx 50% complete for ENCFF097DJJ.fastq\n",
            "Approx 55% complete for ENCFF097DJJ.fastq\n",
            "Approx 60% complete for ENCFF097DJJ.fastq\n",
            "Approx 65% complete for ENCFF097DJJ.fastq\n",
            "Approx 70% complete for ENCFF097DJJ.fastq\n",
            "Approx 75% complete for ENCFF097DJJ.fastq\n",
            "Approx 80% complete for ENCFF097DJJ.fastq\n",
            "Approx 85% complete for ENCFF097DJJ.fastq\n",
            "Approx 90% complete for ENCFF097DJJ.fastq\n",
            "Approx 95% complete for ENCFF097DJJ.fastq\n",
            "Analysis complete for ENCFF097DJJ.fastq\n",
            "Started analysis of ENCFF161DOP.fastq\n",
            "Approx 5% complete for ENCFF161DOP.fastq\n",
            "Approx 10% complete for ENCFF161DOP.fastq\n",
            "Approx 15% complete for ENCFF161DOP.fastq\n",
            "Approx 20% complete for ENCFF161DOP.fastq\n",
            "Approx 25% complete for ENCFF161DOP.fastq\n",
            "Approx 30% complete for ENCFF161DOP.fastq\n",
            "Approx 35% complete for ENCFF161DOP.fastq\n",
            "Approx 40% complete for ENCFF161DOP.fastq\n",
            "Approx 45% complete for ENCFF161DOP.fastq\n",
            "Approx 50% complete for ENCFF161DOP.fastq\n",
            "Approx 55% complete for ENCFF161DOP.fastq\n",
            "Approx 60% complete for ENCFF161DOP.fastq\n",
            "Approx 65% complete for ENCFF161DOP.fastq\n",
            "Approx 70% complete for ENCFF161DOP.fastq\n",
            "Approx 75% complete for ENCFF161DOP.fastq\n",
            "Approx 80% complete for ENCFF161DOP.fastq\n",
            "Approx 85% complete for ENCFF161DOP.fastq\n",
            "Approx 90% complete for ENCFF161DOP.fastq\n",
            "Approx 95% complete for ENCFF161DOP.fastq\n",
            "Analysis complete for ENCFF161DOP.fastq\n",
            "Started analysis of ENCFF259JCU.fastq\n",
            "Approx 5% complete for ENCFF259JCU.fastq\n",
            "Approx 10% complete for ENCFF259JCU.fastq\n",
            "Approx 15% complete for ENCFF259JCU.fastq\n",
            "Approx 20% complete for ENCFF259JCU.fastq\n",
            "Approx 25% complete for ENCFF259JCU.fastq\n",
            "Approx 30% complete for ENCFF259JCU.fastq\n",
            "Approx 35% complete for ENCFF259JCU.fastq\n",
            "Approx 40% complete for ENCFF259JCU.fastq\n",
            "Approx 45% complete for ENCFF259JCU.fastq\n",
            "Approx 50% complete for ENCFF259JCU.fastq\n",
            "Approx 55% complete for ENCFF259JCU.fastq\n",
            "Approx 60% complete for ENCFF259JCU.fastq\n",
            "Approx 65% complete for ENCFF259JCU.fastq\n",
            "Approx 70% complete for ENCFF259JCU.fastq\n",
            "Approx 75% complete for ENCFF259JCU.fastq\n",
            "Approx 80% complete for ENCFF259JCU.fastq\n",
            "Approx 85% complete for ENCFF259JCU.fastq\n",
            "Approx 90% complete for ENCFF259JCU.fastq\n",
            "Approx 95% complete for ENCFF259JCU.fastq\n",
            "Analysis complete for ENCFF259JCU.fastq\n"
          ]
        }
      ],
      "source": [
        "!./FastQC/fastqc /content/ENCFF097DJJ.fastq\n",
        "!./FastQC/fastqc /content/ENCFF161DOP.fastq\n",
        "!./FastQC/fastqc /content/ENCFF259JCU.fastq"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "a06HC3mmNG2u"
      },
      "source": [
        "***\n",
        "\n",
        "Анализ FastQC см. в README.\n",
        "\n",
        "Дальнейшее подрезание прочтений по результатам анализа не потребовалось.\n",
        "\n",
        "***"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "W9QHWBLlpMjQ"
      },
      "source": [
        "## Выравнивание на хромосому\n",
        "\n",
        "Следует выбрать одну хромосому, потому что ресурсы google colab ограничены. В случае проблем, таких как Segmentation fault, error 139, следует выбрать хромосому меньшего размера.\n",
        "\n",
        "Последовательность нуклеотидов хромосомы можно скачать по адресу https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "jF-k2PHnRQds"
      },
      "source": [
        "Возьмем 17 хромосому для выравнивания"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 18,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "N4JIGZOsGiY2",
        "outputId": "dfb64f23-0ab3-4ac2-e86e-14f5e3da8496"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2023-03-06 16:25:09--  https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/chr17.fa.gz\n",
            "Resolving hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)... 128.114.119.163\n",
            "Connecting to hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)|128.114.119.163|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 25930986 (25M) [application/x-gzip]\n",
            "Saving to: ‘chr17.fa.gz’\n",
            "\n",
            "chr17.fa.gz         100%[===================>]  24.73M  11.5MB/s    in 2.2s    \n",
            "\n",
            "2023-03-06 16:25:12 (11.5 MB/s) - ‘chr17.fa.gz’ saved [25930986/25930986]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!wget https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/chr17.fa.gz\n",
        "!gzip -d chr17.fa.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 19,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "bhraQ6NdGiPN",
        "outputId": "0141a14e-3887-4dbf-a78b-3df5d0313715"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Settings:\n",
            "  Output files: \"chromosome_index.*.bt2\"\n",
            "  Line rate: 6 (line is 64 bytes)\n",
            "  Lines per side: 1 (side is 64 bytes)\n",
            "  Offset rate: 4 (one in 16)\n",
            "  FTable chars: 10\n",
            "  Strings: unpacked\n",
            "  Max bucket size: default\n",
            "  Max bucket size, sqrt multiplier: default\n",
            "  Max bucket size, len divisor: 4\n",
            "  Difference-cover sample period: 1024\n",
            "  Endianness: little\n",
            "  Actual local endianness: little\n",
            "  Sanity checking: disabled\n",
            "  Assertions: disabled\n",
            "  Random seed: 0\n",
            "  Sizeofs: void*:8, int:4, long:8, size_t:8\n",
            "Input files DNA, FASTA:\n",
            "  chr17.fa\n",
            "Building a SMALL index\n",
            "Reading reference sizes\n",
            "  Time reading reference sizes: 00:00:01\n",
            "Calculating joined length\n",
            "Writing header\n",
            "Reserving space for joined string\n",
            "Joining reference sequences\n",
            "  Time to join reference sequences: 00:00:01\n",
            "bmax according to bmaxDivN setting: 20730051\n",
            "Using parameters --bmax 15547539 --dcv 1024\n",
            "  Doing ahead-of-time memory usage test\n",
            "  Passed!  Constructing with these parameters: --bmax 15547539 --dcv 1024\n",
            "Constructing suffix-array element generator\n",
            "Building DifferenceCoverSample\n",
            "  Building sPrime\n",
            "  Building sPrimeOrder\n",
            "  V-Sorting samples\n",
            "  V-Sorting samples time: 00:00:03\n",
            "  Allocating rank array\n",
            "  Ranking v-sort output\n",
            "  Ranking v-sort output time: 00:00:01\n",
            "  Invoking Larsson-Sadakane on ranks\n",
            "  Invoking Larsson-Sadakane on ranks time: 00:00:01\n",
            "  Sanity-checking and returning\n",
            "Building samples\n",
            "Reserving space for 12 sample suffixes\n",
            "Generating random suffixes\n",
            "QSorting 12 sample offsets, eliminating duplicates\n",
            "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
            "Multikey QSorting 12 samples\n",
            "  (Using difference cover)\n",
            "  Multikey QSorting samples time: 00:00:00\n",
            "Calculating bucket sizes\n",
            "Splitting and merging\n",
            "  Splitting and merging time: 00:00:00\n",
            "Split 1, merged 6; iterating...\n",
            "Splitting and merging\n",
            "  Splitting and merging time: 00:00:00\n",
            "Avg bucket size: 1.0365e+07 (target: 15547538)\n",
            "Converting suffix-array elements to index image\n",
            "Allocating ftab, absorbFtab\n",
            "Entering Ebwt loop\n",
            "Getting block 1 of 8\n",
            "  Reserving size (15547539) for bucket 1\n",
            "  Calculating Z arrays for bucket 1\n",
            "  Entering block accumulator loop for bucket 1:\n",
            "  bucket 1: 10%\n",
            "  bucket 1: 20%\n",
            "  bucket 1: 30%\n",
            "  bucket 1: 40%\n",
            "  bucket 1: 50%\n",
            "  bucket 1: 60%\n",
            "  bucket 1: 70%\n",
            "  bucket 1: 80%\n",
            "  bucket 1: 90%\n",
            "  bucket 1: 100%\n",
            "  Sorting block of length 6202422 for bucket 1\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:05\n",
            "Returning block of 6202423 for bucket 1\n",
            "Getting block 2 of 8\n",
            "  Reserving size (15547539) for bucket 2\n",
            "  Calculating Z arrays for bucket 2\n",
            "  Entering block accumulator loop for bucket 2:\n",
            "  bucket 2: 10%\n",
            "  bucket 2: 20%\n",
            "  bucket 2: 30%\n",
            "  bucket 2: 40%\n",
            "  bucket 2: 50%\n",
            "  bucket 2: 60%\n",
            "  bucket 2: 70%\n",
            "  bucket 2: 80%\n",
            "  bucket 2: 90%\n",
            "  bucket 2: 100%\n",
            "  Sorting block of length 9523099 for bucket 2\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:08\n",
            "Returning block of 9523100 for bucket 2\n",
            "Getting block 3 of 8\n",
            "  Reserving size (15547539) for bucket 3\n",
            "  Calculating Z arrays for bucket 3\n",
            "  Entering block accumulator loop for bucket 3:\n",
            "  bucket 3: 10%\n",
            "  bucket 3: 20%\n",
            "  bucket 3: 30%\n",
            "  bucket 3: 40%\n",
            "  bucket 3: 50%\n",
            "  bucket 3: 60%\n",
            "  bucket 3: 70%\n",
            "  bucket 3: 80%\n",
            "  bucket 3: 90%\n",
            "  bucket 3: 100%\n",
            "  Sorting block of length 7611579 for bucket 3\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:05\n",
            "Returning block of 7611580 for bucket 3\n",
            "Getting block 4 of 8\n",
            "  Reserving size (15547539) for bucket 4\n",
            "  Calculating Z arrays for bucket 4\n",
            "  Entering block accumulator loop for bucket 4:\n",
            "  bucket 4: 10%\n",
            "  bucket 4: 20%\n",
            "  bucket 4: 30%\n",
            "  bucket 4: 40%\n",
            "  bucket 4: 50%\n",
            "  bucket 4: 60%\n",
            "  bucket 4: 70%\n",
            "  bucket 4: 80%\n",
            "  bucket 4: 90%\n",
            "  bucket 4: 100%\n",
            "  Sorting block of length 9634916 for bucket 4\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:06\n",
            "Returning block of 9634917 for bucket 4\n",
            "Getting block 5 of 8\n",
            "  Reserving size (15547539) for bucket 5\n",
            "  Calculating Z arrays for bucket 5\n",
            "  Entering block accumulator loop for bucket 5:\n",
            "  bucket 5: 10%\n",
            "  bucket 5: 20%\n",
            "  bucket 5: 30%\n",
            "  bucket 5: 40%\n",
            "  bucket 5: 50%\n",
            "  bucket 5: 60%\n",
            "  bucket 5: 70%\n",
            "  bucket 5: 80%\n",
            "  bucket 5: 90%\n",
            "  bucket 5: 100%\n",
            "  Sorting block of length 13344977 for bucket 5\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:09\n",
            "Returning block of 13344978 for bucket 5\n",
            "Getting block 6 of 8\n",
            "  Reserving size (15547539) for bucket 6\n",
            "  Calculating Z arrays for bucket 6\n",
            "  Entering block accumulator loop for bucket 6:\n",
            "  bucket 6: 10%\n",
            "  bucket 6: 20%\n",
            "  bucket 6: 30%\n",
            "  bucket 6: 40%\n",
            "  bucket 6: 50%\n",
            "  bucket 6: 60%\n",
            "  bucket 6: 70%\n",
            "  bucket 6: 80%\n",
            "  bucket 6: 90%\n",
            "  bucket 6: 100%\n",
            "  Sorting block of length 14395028 for bucket 6\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:10\n",
            "Returning block of 14395029 for bucket 6\n",
            "Getting block 7 of 8\n",
            "  Reserving size (15547539) for bucket 7\n",
            "  Calculating Z arrays for bucket 7\n",
            "  Entering block accumulator loop for bucket 7:\n",
            "  bucket 7: 10%\n",
            "  bucket 7: 20%\n",
            "  bucket 7: 30%\n",
            "  bucket 7: 40%\n",
            "  bucket 7: 50%\n",
            "  bucket 7: 60%\n",
            "  bucket 7: 70%\n",
            "  bucket 7: 80%\n",
            "  bucket 7: 90%\n",
            "  bucket 7: 100%\n",
            "  Sorting block of length 12209366 for bucket 7\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:10\n",
            "Returning block of 12209367 for bucket 7\n",
            "Getting block 8 of 8\n",
            "  Reserving size (15547539) for bucket 8\n",
            "  Calculating Z arrays for bucket 8\n",
            "  Entering block accumulator loop for bucket 8:\n",
            "  bucket 8: 10%\n",
            "  bucket 8: 20%\n",
            "  bucket 8: 30%\n",
            "  bucket 8: 40%\n",
            "  bucket 8: 50%\n",
            "  bucket 8: 60%\n",
            "  bucket 8: 70%\n",
            "  bucket 8: 80%\n",
            "  bucket 8: 90%\n",
            "  bucket 8: 100%\n",
            "  Sorting block of length 9998810 for bucket 8\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:07\n",
            "Returning block of 9998811 for bucket 8\n",
            "Exited Ebwt loop\n",
            "fchr[A]: 0\n",
            "fchr[C]: 22639499\n",
            "fchr[G]: 41363443\n",
            "fchr[T]: 60214943\n",
            "fchr[$]: 82920204\n",
            "Exiting Ebwt::buildToDisk()\n",
            "Returning from initFromVector\n",
            "Wrote 31835151 bytes to primary EBWT file: chromosome_index.1.bt2.tmp\n",
            "Wrote 20730056 bytes to secondary EBWT file: chromosome_index.2.bt2.tmp\n",
            "Re-opening _in1 and _in2 as input streams\n",
            "Returning from Ebwt constructor\n",
            "Headers:\n",
            "    len: 82920204\n",
            "    bwtLen: 82920205\n",
            "    sz: 20730051\n",
            "    bwtSz: 20730052\n",
            "    lineRate: 6\n",
            "    offRate: 4\n",
            "    offMask: 0xfffffff0\n",
            "    ftabChars: 10\n",
            "    eftabLen: 20\n",
            "    eftabSz: 80\n",
            "    ftabLen: 1048577\n",
            "    ftabSz: 4194308\n",
            "    offsLen: 5182513\n",
            "    offsSz: 20730052\n",
            "    lineSz: 64\n",
            "    sideSz: 64\n",
            "    sideBwtSz: 48\n",
            "    sideBwtLen: 192\n",
            "    numSides: 431877\n",
            "    numLines: 431877\n",
            "    ebwtTotLen: 27640128\n",
            "    ebwtTotSz: 27640128\n",
            "    color: 0\n",
            "    reverse: 0\n",
            "Total time for call to driver() for forward index: 00:01:49\n",
            "Reading reference sizes\n",
            "  Time reading reference sizes: 00:00:01\n",
            "Calculating joined length\n",
            "Writing header\n",
            "Reserving space for joined string\n",
            "Joining reference sequences\n",
            "  Time to join reference sequences: 00:00:01\n",
            "  Time to reverse reference sequence: 00:00:00\n",
            "bmax according to bmaxDivN setting: 20730051\n",
            "Using parameters --bmax 15547539 --dcv 1024\n",
            "  Doing ahead-of-time memory usage test\n",
            "  Passed!  Constructing with these parameters: --bmax 15547539 --dcv 1024\n",
            "Constructing suffix-array element generator\n",
            "Building DifferenceCoverSample\n",
            "  Building sPrime\n",
            "  Building sPrimeOrder\n",
            "  V-Sorting samples\n",
            "  V-Sorting samples time: 00:00:03\n",
            "  Allocating rank array\n",
            "  Ranking v-sort output\n",
            "  Ranking v-sort output time: 00:00:01\n",
            "  Invoking Larsson-Sadakane on ranks\n",
            "  Invoking Larsson-Sadakane on ranks time: 00:00:01\n",
            "  Sanity-checking and returning\n",
            "Building samples\n",
            "Reserving space for 12 sample suffixes\n",
            "Generating random suffixes\n",
            "QSorting 12 sample offsets, eliminating duplicates\n",
            "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
            "Multikey QSorting 12 samples\n",
            "  (Using difference cover)\n",
            "  Multikey QSorting samples time: 00:00:00\n",
            "Calculating bucket sizes\n",
            "Splitting and merging\n",
            "  Splitting and merging time: 00:00:00\n",
            "Split 2, merged 6; iterating...\n",
            "Splitting and merging\n",
            "  Splitting and merging time: 00:00:00\n",
            "Split 1, merged 1; iterating...\n",
            "Splitting and merging\n",
            "  Splitting and merging time: 00:00:00\n",
            "Avg bucket size: 9.21336e+06 (target: 15547538)\n",
            "Converting suffix-array elements to index image\n",
            "Allocating ftab, absorbFtab\n",
            "Entering Ebwt loop\n",
            "Getting block 1 of 9\n",
            "  Reserving size (15547539) for bucket 1\n",
            "  Calculating Z arrays for bucket 1\n",
            "  Entering block accumulator loop for bucket 1:\n",
            "  bucket 1: 10%\n",
            "  bucket 1: 20%\n",
            "  bucket 1: 30%\n",
            "  bucket 1: 40%\n",
            "  bucket 1: 50%\n",
            "  bucket 1: 60%\n",
            "  bucket 1: 70%\n",
            "  bucket 1: 80%\n",
            "  bucket 1: 90%\n",
            "  bucket 1: 100%\n",
            "  Sorting block of length 11229738 for bucket 1\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:08\n",
            "Returning block of 11229739 for bucket 1\n",
            "Getting block 2 of 9\n",
            "  Reserving size (15547539) for bucket 2\n",
            "  Calculating Z arrays for bucket 2\n",
            "  Entering block accumulator loop for bucket 2:\n",
            "  bucket 2: 10%\n",
            "  bucket 2: 20%\n",
            "  bucket 2: 30%\n",
            "  bucket 2: 40%\n",
            "  bucket 2: 50%\n",
            "  bucket 2: 60%\n",
            "  bucket 2: 70%\n",
            "  bucket 2: 80%\n",
            "  bucket 2: 90%\n",
            "  bucket 2: 100%\n",
            "  Sorting block of length 7096767 for bucket 2\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:06\n",
            "Returning block of 7096768 for bucket 2\n",
            "Getting block 3 of 9\n",
            "  Reserving size (15547539) for bucket 3\n",
            "  Calculating Z arrays for bucket 3\n",
            "  Entering block accumulator loop for bucket 3:\n",
            "  bucket 3: 10%\n",
            "  bucket 3: 20%\n",
            "  bucket 3: 30%\n",
            "  bucket 3: 40%\n",
            "  bucket 3: 50%\n",
            "  bucket 3: 60%\n",
            "  bucket 3: 70%\n",
            "  bucket 3: 80%\n",
            "  bucket 3: 90%\n",
            "  bucket 3: 100%\n",
            "  Sorting block of length 12893781 for bucket 3\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:08\n",
            "Returning block of 12893782 for bucket 3\n",
            "Getting block 4 of 9\n",
            "  Reserving size (15547539) for bucket 4\n",
            "  Calculating Z arrays for bucket 4\n",
            "  Entering block accumulator loop for bucket 4:\n",
            "  bucket 4: 10%\n",
            "  bucket 4: 20%\n",
            "  bucket 4: 30%\n",
            "  bucket 4: 40%\n",
            "  bucket 4: 50%\n",
            "  bucket 4: 60%\n",
            "  bucket 4: 70%\n",
            "  bucket 4: 80%\n",
            "  bucket 4: 90%\n",
            "  bucket 4: 100%\n",
            "  Sorting block of length 5914851 for bucket 4\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:05\n",
            "Returning block of 5914852 for bucket 4\n",
            "Getting block 5 of 9\n",
            "  Reserving size (15547539) for bucket 5\n",
            "  Calculating Z arrays for bucket 5\n",
            "  Entering block accumulator loop for bucket 5:\n",
            "  bucket 5: 10%\n",
            "  bucket 5: 20%\n",
            "  bucket 5: 30%\n",
            "  bucket 5: 40%\n",
            "  bucket 5: 50%\n",
            "  bucket 5: 60%\n",
            "  bucket 5: 70%\n",
            "  bucket 5: 80%\n",
            "  bucket 5: 90%\n",
            "  bucket 5: 100%\n",
            "  Sorting block of length 9847183 for bucket 5\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:07\n",
            "Returning block of 9847184 for bucket 5\n",
            "Getting block 6 of 9\n",
            "  Reserving size (15547539) for bucket 6\n",
            "  Calculating Z arrays for bucket 6\n",
            "  Entering block accumulator loop for bucket 6:\n",
            "  bucket 6: 10%\n",
            "  bucket 6: 20%\n",
            "  bucket 6: 30%\n",
            "  bucket 6: 40%\n",
            "  bucket 6: 50%\n",
            "  bucket 6: 60%\n",
            "  bucket 6: 70%\n",
            "  bucket 6: 80%\n",
            "  bucket 6: 90%\n",
            "  bucket 6: 100%\n",
            "  Sorting block of length 11197000 for bucket 6\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:08\n",
            "Returning block of 11197001 for bucket 6\n",
            "Getting block 7 of 9\n",
            "  Reserving size (15547539) for bucket 7\n",
            "  Calculating Z arrays for bucket 7\n",
            "  Entering block accumulator loop for bucket 7:\n",
            "  bucket 7: 10%\n",
            "  bucket 7: 20%\n",
            "  bucket 7: 30%\n",
            "  bucket 7: 40%\n",
            "  bucket 7: 50%\n",
            "  bucket 7: 60%\n",
            "  bucket 7: 70%\n",
            "  bucket 7: 80%\n",
            "  bucket 7: 90%\n",
            "  bucket 7: 100%\n",
            "  Sorting block of length 8171042 for bucket 7\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:07\n",
            "Returning block of 8171043 for bucket 7\n",
            "Getting block 8 of 9\n",
            "  Reserving size (15547539) for bucket 8\n",
            "  Calculating Z arrays for bucket 8\n",
            "  Entering block accumulator loop for bucket 8:\n",
            "  bucket 8: 10%\n",
            "  bucket 8: 20%\n",
            "  bucket 8: 30%\n",
            "  bucket 8: 40%\n",
            "  bucket 8: 50%\n",
            "  bucket 8: 60%\n",
            "  bucket 8: 70%\n",
            "  bucket 8: 80%\n",
            "  bucket 8: 90%\n",
            "  bucket 8: 100%\n",
            "  Sorting block of length 14639375 for bucket 8\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:10\n",
            "Returning block of 14639376 for bucket 8\n",
            "Getting block 9 of 9\n",
            "  Reserving size (15547539) for bucket 9\n",
            "  Calculating Z arrays for bucket 9\n",
            "  Entering block accumulator loop for bucket 9:\n",
            "  bucket 9: 10%\n",
            "  bucket 9: 20%\n",
            "  bucket 9: 30%\n",
            "  bucket 9: 40%\n",
            "  bucket 9: 50%\n",
            "  bucket 9: 60%\n",
            "  bucket 9: 70%\n",
            "  bucket 9: 80%\n",
            "  bucket 9: 90%\n",
            "  bucket 9: 100%\n",
            "  Sorting block of length 1930459 for bucket 9\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:01\n",
            "Returning block of 1930460 for bucket 9\n",
            "Exited Ebwt loop\n",
            "fchr[A]: 0\n",
            "fchr[C]: 22639499\n",
            "fchr[G]: 41363443\n",
            "fchr[T]: 60214943\n",
            "fchr[$]: 82920204\n",
            "Exiting Ebwt::buildToDisk()\n",
            "Returning from initFromVector\n",
            "Wrote 31835151 bytes to primary EBWT file: chromosome_index.rev.1.bt2.tmp\n",
            "Wrote 20730056 bytes to secondary EBWT file: chromosome_index.rev.2.bt2.tmp\n",
            "Re-opening _in1 and _in2 as input streams\n",
            "Returning from Ebwt constructor\n",
            "Headers:\n",
            "    len: 82920204\n",
            "    bwtLen: 82920205\n",
            "    sz: 20730051\n",
            "    bwtSz: 20730052\n",
            "    lineRate: 6\n",
            "    offRate: 4\n",
            "    offMask: 0xfffffff0\n",
            "    ftabChars: 10\n",
            "    eftabLen: 20\n",
            "    eftabSz: 80\n",
            "    ftabLen: 1048577\n",
            "    ftabSz: 4194308\n",
            "    offsLen: 5182513\n",
            "    offsSz: 20730052\n",
            "    lineSz: 64\n",
            "    sideSz: 64\n",
            "    sideBwtSz: 48\n",
            "    sideBwtLen: 192\n",
            "    numSides: 431877\n",
            "    numLines: 431877\n",
            "    ebwtTotLen: 27640128\n",
            "    ebwtTotSz: 27640128\n",
            "    color: 0\n",
            "    reverse: 1\n",
            "Total time for backward call to driver() for mirror index: 00:01:55\n",
            "Renaming chromosome_index.3.bt2.tmp to chromosome_index.3.bt2\n",
            "Renaming chromosome_index.4.bt2.tmp to chromosome_index.4.bt2\n",
            "Renaming chromosome_index.1.bt2.tmp to chromosome_index.1.bt2\n",
            "Renaming chromosome_index.2.bt2.tmp to chromosome_index.2.bt2\n",
            "Renaming chromosome_index.rev.1.bt2.tmp to chromosome_index.rev.1.bt2\n",
            "Renaming chromosome_index.rev.2.bt2.tmp to chromosome_index.rev.2.bt2\n",
            "CPU times: user 2.35 s, sys: 250 ms, total: 2.6 s\n",
            "Wall time: 3min 44s\n"
          ]
        }
      ],
      "source": [
        "%%time\n",
        "!bowtie2-build chr17.fa chromosome_index"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 20,
      "metadata": {
        "id": "E_usI8XUrQnR"
      },
      "outputs": [],
      "source": [
        "!mkdir bowtie2_res"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 21,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "3t80UATHqFe7",
        "outputId": "ba890282-7c66-4f57-ce12-ab66df580db0"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "34403162 reads; of these:\n",
            "  34403162 (100.00%) were unpaired; of these:\n",
            "    29299462 (85.17%) aligned 0 times\n",
            "    1222087 (3.55%) aligned exactly 1 time\n",
            "    3881613 (11.28%) aligned >1 times\n",
            "14.83% overall alignment rate\n",
            "CPU times: user 21.4 s, sys: 1.87 s, total: 23.3 s\n",
            "Wall time: 36min 47s\n"
          ]
        }
      ],
      "source": [
        "# 1 реплика\n",
        "%%time\n",
        "!bowtie2  -p 2 \\\n",
        "          -x chromosome_index \\\n",
        "          -U ENCFF097DJJ.fastq \\\n",
        "          -S bowtie2_res/res_DJJ.sam"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 22,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "so1jTdGHoEir",
        "outputId": "cf70044c-a9fd-48fa-a80a-e13c5848f2af"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "64763664 reads; of these:\n",
            "  64763664 (100.00%) were unpaired; of these:\n",
            "    55166245 (85.18%) aligned 0 times\n",
            "    2657042 (4.10%) aligned exactly 1 time\n",
            "    6940377 (10.72%) aligned >1 times\n",
            "14.82% overall alignment rate\n",
            "CPU times: user 40.2 s, sys: 3.41 s, total: 43.6 s\n",
            "Wall time: 1h 8min 33s\n"
          ]
        }
      ],
      "source": [
        "# 2 реплика\n",
        "%%time\n",
        "!bowtie2  -p 2 \\\n",
        "          -x chromosome_index \\\n",
        "          -U ENCFF161DOP.fastq \\\n",
        "          -S bowtie2_res/res_DOP.sam"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 23,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "l4PxypR6SjZP",
        "outputId": "ab52e889-a064-491d-f20b-9dc6f09a29d0"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "88755850 reads; of these:\n",
            "  88755850 (100.00%) were unpaired; of these:\n",
            "    74376714 (83.80%) aligned 0 times\n",
            "    3302303 (3.72%) aligned exactly 1 time\n",
            "    11076833 (12.48%) aligned >1 times\n",
            "16.20% overall alignment rate\n",
            "CPU times: user 58.1 s, sys: 5.12 s, total: 1min 3s\n",
            "Wall time: 1h 40min 51s\n"
          ]
        }
      ],
      "source": [
        "# контроль\n",
        "%%time\n",
        "!bowtie2  -p 2 \\\n",
        "          -x chromosome_index \\\n",
        "          -U ENCFF259JCU.fastq \\\n",
        "          -S bowtie2_res/res_JCU.sam"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "jZJiaaJBZBc7"
      },
      "source": [
        "### Таблица со статистикой по выравниванию"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 24,
      "metadata": {
        "id": "RvQFuOhMXcw3"
      },
      "outputs": [],
      "source": [
        "import pandas as pd"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 25,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 222
        },
        "id": "zp118ckhXffz",
        "outputId": "821e9aa0-2073-4b46-bfc7-bd5eb579de95"
      },
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "             Общее число ридов Выровнившиеся уникально  \\\n",
              "ENCFF097DJJ           34403162                 1222087   \n",
              "ENCFF161DOP           64763664                 2657042   \n",
              "ENCFF259JCU           88755850                 2657042   \n",
              "\n",
              "            Выровнившиеся уникально (%) Выровнившиеся неуникально  \\\n",
              "ENCFF097DJJ                       3.55%                   3881613   \n",
              "ENCFF161DOP                       4.10%                   6940377   \n",
              "ENCFF259JCU                       3.72%                  11076833   \n",
              "\n",
              "            Выровнившиеся неуникально (%) Не выровнившиеся  \\\n",
              "ENCFF097DJJ                        11.28%         29299462   \n",
              "ENCFF161DOP                        10.72%         55166245   \n",
              "ENCFF259JCU                        12.48%         74376714   \n",
              "\n",
              "            Не выровнившиеся (%)  \n",
              "ENCFF097DJJ               85.17%  \n",
              "ENCFF161DOP               85.18%  \n",
              "ENCFF259JCU               83.80%  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-7bfe6d22-746b-48f0-9bee-1b43b6e566f0\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>Общее число ридов</th>\n",
              "      <th>Выровнившиеся уникально</th>\n",
              "      <th>Выровнившиеся уникально (%)</th>\n",
              "      <th>Выровнившиеся неуникально</th>\n",
              "      <th>Выровнившиеся неуникально (%)</th>\n",
              "      <th>Не выровнившиеся</th>\n",
              "      <th>Не выровнившиеся (%)</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>ENCFF097DJJ</th>\n",
              "      <td>34403162</td>\n",
              "      <td>1222087</td>\n",
              "      <td>3.55%</td>\n",
              "      <td>3881613</td>\n",
              "      <td>11.28%</td>\n",
              "      <td>29299462</td>\n",
              "      <td>85.17%</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>ENCFF161DOP</th>\n",
              "      <td>64763664</td>\n",
              "      <td>2657042</td>\n",
              "      <td>4.10%</td>\n",
              "      <td>6940377</td>\n",
              "      <td>10.72%</td>\n",
              "      <td>55166245</td>\n",
              "      <td>85.18%</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>ENCFF259JCU</th>\n",
              "      <td>88755850</td>\n",
              "      <td>2657042</td>\n",
              "      <td>3.72%</td>\n",
              "      <td>11076833</td>\n",
              "      <td>12.48%</td>\n",
              "      <td>74376714</td>\n",
              "      <td>83.80%</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-7bfe6d22-746b-48f0-9bee-1b43b6e566f0')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-7bfe6d22-746b-48f0-9bee-1b43b6e566f0 button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-7bfe6d22-746b-48f0-9bee-1b43b6e566f0');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ]
          },
          "metadata": {},
          "execution_count": 25
        }
      ],
      "source": [
        "data = {'Общее число ридов':[34403162,\n",
        "                             64763664,\n",
        "                             88755850],\n",
        "        'Выровнившиеся уникально':['1222087',\n",
        "                                   '2657042',\n",
        "                                   '2657042'],\n",
        "        'Выровнившиеся уникально (%)':['3.55%', \n",
        "                                       '4.10%',\n",
        "                                       '3.72%'],\n",
        "        'Выровнившиеся неуникально':['3881613',\n",
        "                                     '6940377',\n",
        "                                     '11076833'],\n",
        "        'Выровнившиеся неуникально (%)':['11.28%',\n",
        "                                         '10.72%',\n",
        "                                         '12.48%'],\n",
        "        'Не выровнившиеся':['29299462',\n",
        "                            '55166245',\n",
        "                            '74376714'],\n",
        "        'Не выровнившиеся (%)':['85.17%',\n",
        "                                '85.18%',\n",
        "                                '83.80%']}\n",
        "\n",
        "df = pd.DataFrame(data, index =['ENCFF097DJJ',\n",
        "                                'ENCFF161DOP',\n",
        "                                'ENCFF259JCU'])\n",
        "df"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 26,
      "metadata": {
        "id": "huJ6P_NmZFtO"
      },
      "outputs": [],
      "source": [
        "df.to_csv('stats.csv')"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "ysoDrnSCrei0"
      },
      "source": [
        "Проанализируйте выдачу bowtie. Почему процент выравниваний получился именно таким?\n",
        "\n",
        "***\n",
        "\n",
        "**Ответ:** потому что мы выравнивали риды на одну хромосому, которая составляет малую часть генома человека, ведь там всего 23 пары хромосом. Поэтому процент выравниваний и получился таким низким.\n",
        "\n",
        "***\n",
        "\n",
        "Имеет смысл для дальнейшего анализа отобрать уникально картированные риды."
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "TmfJKT0rmLyp"
      },
      "source": [
        "Преобразование .sam в .bam (не требуется если запускать macs2 на .sam файлах)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 27,
      "metadata": {
        "id": "2PteJ-KVqFhs"
      },
      "outputs": [],
      "source": [
        "!samtools view -h -S -b -o bowtie2_res/res_DJJ.bam bowtie2_res/res_DJJ.sam\n",
        "!samtools view -h -S -b -o bowtie2_res/res_DOP.bam bowtie2_res/res_DOP.sam\n",
        "!samtools view -h -S -b -o bowtie2_res/res_JCU.bam bowtie2_res/res_JCU.sam"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "eLtG7xtRup19"
      },
      "source": [
        "На этом моменте можно скачать себе на компьютер все полученные .bam или .sam файлы на случай сбоя."
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "XZtQrPZMmgnC"
      },
      "source": [
        "## Peak calling\n",
        "\n",
        "Помимо macs2 может быть использован пакет Homer для поиска пиков, а также macs3."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 28,
      "metadata": {
        "id": "uX1ps4VDp_Ae"
      },
      "outputs": [],
      "source": [
        "!mkdir macs2"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!macs2 callpeak --broad -t bowtie2_res/res_DJJ.sam \\\n",
        "    -c bowtie2_res/res_JCU.sam \\\n",
        " \t  -f SAM \\\n",
        "\t  --outdir macs2 \\\n",
        "      -n ENCFF097DJJ"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "j6Iv0r7ysMtf",
        "outputId": "30759333-fd61-4d7d-d92d-19191857e983"
      },
      "execution_count": 70,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "INFO  @ Mon, 06 Mar 2023 20:51:26: \n",
            "# Command line: callpeak --broad -t bowtie2_res/res_DJJ.sam -c bowtie2_res/res_JCU.sam -f SAM --outdir macs2 -n ENCFF097DJJ\n",
            "# ARGUMENTS LIST:\n",
            "# name = ENCFF097DJJ\n",
            "# format = SAM\n",
            "# ChIP-seq file = ['bowtie2_res/res_DJJ.sam']\n",
            "# control file = ['bowtie2_res/res_JCU.sam']\n",
            "# effective genome size = 2.70e+09\n",
            "# band width = 300\n",
            "# model fold = [5, 50]\n",
            "# qvalue cutoff for narrow/strong regions = 5.00e-02\n",
            "# qvalue cutoff for broad/weak regions = 1.00e-01\n",
            "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
            "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
            "# Larger dataset will be scaled towards smaller dataset.\n",
            "# Range for calculating regional lambda is: 1000 bps and 10000 bps\n",
            "# Broad region calling is on\n",
            "# Paired-End mode is off\n",
            " \n",
            "INFO  @ Mon, 06 Mar 2023 20:51:26: #1 read tag files... \n",
            "INFO  @ Mon, 06 Mar 2023 20:51:26: #1 read treatment tags... \n",
            "INFO  @ Mon, 06 Mar 2023 20:51:30:  3000000 \n",
            "INFO  @ Mon, 06 Mar 2023 20:51:40:  10000000 \n",
            "INFO  @ Mon, 06 Mar 2023 20:51:46:  13000000 \n",
            "INFO  @ Mon, 06 Mar 2023 20:52:03:  25000000 \n",
            "INFO  @ Mon, 06 Mar 2023 20:52:09:  28000000 \n",
            "INFO  @ Mon, 06 Mar 2023 20:52:12:  30000000 \n",
            "INFO  @ Mon, 06 Mar 2023 20:52:16:  33000000 \n",
            "INFO  @ Mon, 06 Mar 2023 20:52:17:  34000000 \n",
            "INFO  @ Mon, 06 Mar 2023 20:52:18: #1.2 read input tags... \n",
            "INFO  @ Mon, 06 Mar 2023 20:52:23:  5000000 \n",
            "INFO  @ Mon, 06 Mar 2023 20:52:54:  23000000 \n",
            "INFO  @ Mon, 06 Mar 2023 20:53:04:  30000000 \n",
            "INFO  @ Mon, 06 Mar 2023 20:53:15:  38000000 \n",
            "INFO  @ Mon, 06 Mar 2023 20:53:32:  51000000 \n",
            "INFO  @ Mon, 06 Mar 2023 20:53:34:  52000000 \n",
            "INFO  @ Mon, 06 Mar 2023 20:53:45:  59000000 \n",
            "INFO  @ Mon, 06 Mar 2023 20:53:52:  66000000 \n",
            "INFO  @ Mon, 06 Mar 2023 20:54:09:  77000000 \n",
            "INFO  @ Mon, 06 Mar 2023 20:54:10:  78000000 \n",
            "INFO  @ Mon, 06 Mar 2023 20:54:16:  83000000 \n",
            "INFO  @ Mon, 06 Mar 2023 20:54:25:  88000000 \n",
            "INFO  @ Mon, 06 Mar 2023 20:54:28: #1 tag size is determined as 50 bps \n",
            "INFO  @ Mon, 06 Mar 2023 20:54:28: #1 tag size = 50.0 \n",
            "INFO  @ Mon, 06 Mar 2023 20:54:28: #1  total tags in treatment: 5103700 \n",
            "INFO  @ Mon, 06 Mar 2023 20:54:28: #1 user defined the maximum tags... \n",
            "INFO  @ Mon, 06 Mar 2023 20:54:28: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
            "INFO  @ Mon, 06 Mar 2023 20:54:28: #1  tags after filtering in treatment: 3836826 \n",
            "INFO  @ Mon, 06 Mar 2023 20:54:28: #1  Redundant rate of treatment: 0.25 \n",
            "INFO  @ Mon, 06 Mar 2023 20:54:28: #1  total tags in control: 14379136 \n",
            "INFO  @ Mon, 06 Mar 2023 20:54:28: #1 user defined the maximum tags... \n",
            "INFO  @ Mon, 06 Mar 2023 20:54:28: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
            "INFO  @ Mon, 06 Mar 2023 20:54:28: #1  tags after filtering in control: 8827819 \n",
            "INFO  @ Mon, 06 Mar 2023 20:54:28: #1  Redundant rate of control: 0.39 \n",
            "INFO  @ Mon, 06 Mar 2023 20:54:28: #1 finished! \n",
            "INFO  @ Mon, 06 Mar 2023 20:54:28: #2 Build Peak Model... \n",
            "INFO  @ Mon, 06 Mar 2023 20:54:28: #2 looking for paired plus/minus strand peaks... \n",
            "INFO  @ Mon, 06 Mar 2023 20:54:30: #2 number of paired peaks: 37409 \n",
            "INFO  @ Mon, 06 Mar 2023 20:54:30: start model_add_line... \n",
            "INFO  @ Mon, 06 Mar 2023 20:54:30: start X-correlation... \n",
            "INFO  @ Mon, 06 Mar 2023 20:54:30: end of X-cor \n",
            "INFO  @ Mon, 06 Mar 2023 20:54:30: #2 finished! \n",
            "INFO  @ Mon, 06 Mar 2023 20:54:30: #2 predicted fragment length is 50 bps \n",
            "INFO  @ Mon, 06 Mar 2023 20:54:30: #2 alternative fragment length(s) may be 50 bps \n",
            "INFO  @ Mon, 06 Mar 2023 20:54:30: #2.2 Generate R script for model : macs2/ENCFF097DJJ_model.r \n",
            "WARNING @ Mon, 06 Mar 2023 20:54:30: #2 Since the d (50) calculated from paired-peaks are smaller than 2*tag length, it may be influenced by unknown sequencing problem! \n",
            "WARNING @ Mon, 06 Mar 2023 20:54:30: #2 You may need to consider one of the other alternative d(s): 50 \n",
            "WARNING @ Mon, 06 Mar 2023 20:54:30: #2 You can restart the process with --nomodel --extsize XXX with your choice or an arbitrary number. Nontheless, MACS will continute computing. \n",
            "INFO  @ Mon, 06 Mar 2023 20:54:30: #3 Call peaks... \n",
            "INFO  @ Mon, 06 Mar 2023 20:54:30: #3 Call broad peaks with given level1 -log10qvalue cutoff and level2: 1.301030, 1.000000... \n",
            "INFO  @ Mon, 06 Mar 2023 20:54:30: #3 Pre-compute pvalue-qvalue table... \n",
            "INFO  @ Mon, 06 Mar 2023 20:55:17: #3 Call peaks for each chromosome... \n",
            "INFO  @ Mon, 06 Mar 2023 20:55:41: #4 Write output xls file... macs2/ENCFF097DJJ_peaks.xls \n",
            "INFO  @ Mon, 06 Mar 2023 20:55:41: #4 Write broad peak in broadPeak format file... macs2/ENCFF097DJJ_peaks.broadPeak \n",
            "INFO  @ Mon, 06 Mar 2023 20:55:41: #4 Write broad peak in bed12/gappedPeak format file... macs2/ENCFF097DJJ_peaks.gappedPeak \n",
            "INFO  @ Mon, 06 Mar 2023 20:55:41: Done! \n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 71,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "UiL_pwRpfx7R",
        "outputId": "503c627c-7d36-4630-f295-624e1e0668bb"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "INFO  @ Mon, 06 Mar 2023 20:55:42: \n",
            "# Command line: callpeak --broad -t bowtie2_res/res_DOP.sam -c bowtie2_res/res_JCU.sam -f SAM --outdir macs2 -n ENCFF161DOP\n",
            "# ARGUMENTS LIST:\n",
            "# name = ENCFF161DOP\n",
            "# format = SAM\n",
            "# ChIP-seq file = ['bowtie2_res/res_DOP.sam']\n",
            "# control file = ['bowtie2_res/res_JCU.sam']\n",
            "# effective genome size = 2.70e+09\n",
            "# band width = 300\n",
            "# model fold = [5, 50]\n",
            "# qvalue cutoff for narrow/strong regions = 5.00e-02\n",
            "# qvalue cutoff for broad/weak regions = 1.00e-01\n",
            "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
            "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
            "# Larger dataset will be scaled towards smaller dataset.\n",
            "# Range for calculating regional lambda is: 1000 bps and 10000 bps\n",
            "# Broad region calling is on\n",
            "# Paired-End mode is off\n",
            " \n",
            "INFO  @ Mon, 06 Mar 2023 20:55:42: #1 read tag files... \n",
            "INFO  @ Mon, 06 Mar 2023 20:55:42: #1 read treatment tags... \n",
            "INFO  @ Mon, 06 Mar 2023 20:55:52:  6000000 \n",
            "INFO  @ Mon, 06 Mar 2023 20:55:57:  10000000 \n",
            "INFO  @ Mon, 06 Mar 2023 20:56:09:  19000000 \n",
            "INFO  @ Mon, 06 Mar 2023 20:56:24:  27000000 \n",
            "INFO  @ Mon, 06 Mar 2023 20:56:35:  31000000 \n",
            "INFO  @ Mon, 06 Mar 2023 20:56:44:  35000000 \n",
            "INFO  @ Mon, 06 Mar 2023 20:56:46:  36000000 \n",
            "INFO  @ Mon, 06 Mar 2023 20:56:51:  39000000 \n",
            "INFO  @ Mon, 06 Mar 2023 20:57:20:  59000000 \n",
            "INFO  @ Mon, 06 Mar 2023 20:57:26:  63000000 \n",
            "INFO  @ Mon, 06 Mar 2023 20:57:29: #1.2 read input tags... \n",
            "INFO  @ Mon, 06 Mar 2023 20:57:35:  5000000 \n",
            "INFO  @ Mon, 06 Mar 2023 20:58:01:  23000000 \n",
            "INFO  @ Mon, 06 Mar 2023 20:58:12:  30000000 \n",
            "INFO  @ Mon, 06 Mar 2023 20:58:22:  38000000 \n",
            "INFO  @ Mon, 06 Mar 2023 20:58:40:  51000000 \n",
            "INFO  @ Mon, 06 Mar 2023 20:58:41:  52000000 \n",
            "INFO  @ Mon, 06 Mar 2023 20:58:53:  59000000 \n",
            "INFO  @ Mon, 06 Mar 2023 20:59:01:  66000000 \n",
            "INFO  @ Mon, 06 Mar 2023 20:59:18:  77000000 \n",
            "INFO  @ Mon, 06 Mar 2023 20:59:19:  78000000 \n",
            "INFO  @ Mon, 06 Mar 2023 20:59:25:  83000000 \n",
            "INFO  @ Mon, 06 Mar 2023 20:59:33:  88000000 \n",
            "INFO  @ Mon, 06 Mar 2023 20:59:36: #1 tag size is determined as 50 bps \n",
            "INFO  @ Mon, 06 Mar 2023 20:59:36: #1 tag size = 50.0 \n",
            "INFO  @ Mon, 06 Mar 2023 20:59:36: #1  total tags in treatment: 9597419 \n",
            "INFO  @ Mon, 06 Mar 2023 20:59:36: #1 user defined the maximum tags... \n",
            "INFO  @ Mon, 06 Mar 2023 20:59:36: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
            "INFO  @ Mon, 06 Mar 2023 20:59:37: #1  tags after filtering in treatment: 5881270 \n",
            "INFO  @ Mon, 06 Mar 2023 20:59:37: #1  Redundant rate of treatment: 0.39 \n",
            "INFO  @ Mon, 06 Mar 2023 20:59:37: #1  total tags in control: 14379136 \n",
            "INFO  @ Mon, 06 Mar 2023 20:59:37: #1 user defined the maximum tags... \n",
            "INFO  @ Mon, 06 Mar 2023 20:59:37: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
            "INFO  @ Mon, 06 Mar 2023 20:59:37: #1  tags after filtering in control: 8827819 \n",
            "INFO  @ Mon, 06 Mar 2023 20:59:37: #1  Redundant rate of control: 0.39 \n",
            "INFO  @ Mon, 06 Mar 2023 20:59:37: #1 finished! \n",
            "INFO  @ Mon, 06 Mar 2023 20:59:37: #2 Build Peak Model... \n",
            "INFO  @ Mon, 06 Mar 2023 20:59:37: #2 looking for paired plus/minus strand peaks... \n",
            "INFO  @ Mon, 06 Mar 2023 20:59:39: #2 number of paired peaks: 42223 \n",
            "INFO  @ Mon, 06 Mar 2023 20:59:39: start model_add_line... \n",
            "INFO  @ Mon, 06 Mar 2023 20:59:39: start X-correlation... \n",
            "INFO  @ Mon, 06 Mar 2023 20:59:39: end of X-cor \n",
            "INFO  @ Mon, 06 Mar 2023 20:59:39: #2 finished! \n",
            "INFO  @ Mon, 06 Mar 2023 20:59:39: #2 predicted fragment length is 50 bps \n",
            "INFO  @ Mon, 06 Mar 2023 20:59:39: #2 alternative fragment length(s) may be 50 bps \n",
            "INFO  @ Mon, 06 Mar 2023 20:59:39: #2.2 Generate R script for model : macs2/ENCFF161DOP_model.r \n",
            "WARNING @ Mon, 06 Mar 2023 20:59:39: #2 Since the d (50) calculated from paired-peaks are smaller than 2*tag length, it may be influenced by unknown sequencing problem! \n",
            "WARNING @ Mon, 06 Mar 2023 20:59:39: #2 You may need to consider one of the other alternative d(s): 50 \n",
            "WARNING @ Mon, 06 Mar 2023 20:59:39: #2 You can restart the process with --nomodel --extsize XXX with your choice or an arbitrary number. Nontheless, MACS will continute computing. \n",
            "INFO  @ Mon, 06 Mar 2023 20:59:39: #3 Call peaks... \n",
            "INFO  @ Mon, 06 Mar 2023 20:59:39: #3 Call broad peaks with given level1 -log10qvalue cutoff and level2: 1.301030, 1.000000... \n",
            "INFO  @ Mon, 06 Mar 2023 20:59:39: #3 Pre-compute pvalue-qvalue table... \n",
            "INFO  @ Mon, 06 Mar 2023 21:00:30: #3 Call peaks for each chromosome... \n",
            "INFO  @ Mon, 06 Mar 2023 21:01:07: #4 Write output xls file... macs2/ENCFF161DOP_peaks.xls \n",
            "INFO  @ Mon, 06 Mar 2023 21:01:07: #4 Write broad peak in broadPeak format file... macs2/ENCFF161DOP_peaks.broadPeak \n",
            "INFO  @ Mon, 06 Mar 2023 21:01:07: #4 Write broad peak in bed12/gappedPeak format file... macs2/ENCFF161DOP_peaks.gappedPeak \n",
            "INFO  @ Mon, 06 Mar 2023 21:01:07: Done! \n"
          ]
        }
      ],
      "source": [
        "!macs2 callpeak --broad -t bowtie2_res/res_DOP.sam \\\n",
        "    -c bowtie2_res/res_JCU.sam \\\n",
        " \t  -f SAM \\\n",
        "\t  --outdir macs2 \\\n",
        "      -n ENCFF161DOP"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 31,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "HrNwxs7ArPz7",
        "outputId": "6e507fa5-308d-49bd-8ad1-203d0cf1a158"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2023-03-06 20:12:59--  https://www.encodeproject.org/files/ENCFF428QYO/@@download/ENCFF428QYO.bed.gz\n",
            "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
            "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
            "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
            "Location: https://encode-public.s3.amazonaws.com/2022/01/28/868679f8-fcf5-4906-99b8-4ca0cb1ae34c/ENCFF428QYO.bed.gz?response-content-disposition=attachment%3B%20filename%3DENCFF428QYO.bed.gz&AWSAccessKeyId=ASIATGZNGCNXVHVJY3UO&Signature=Gl9%2FQQhluAZMOipz5nsxs9TdXEo%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEGQaCXVzLXdlc3QtMiJHMEUCIQDNs42kNiu5GM7vFaLtCNH6DYbW4gmTJYbs7HbZtI2qCAIgPoR6U24dak4OrlXiZMnlnzGTAKBFJNtY1WYICAgc4PsqsgUIHRAAGgwyMjA3NDg3MTQ4NjMiDPmqUWnJllxtha8p2CqPBftS7XGYpMYCIHe39YMCLIdoYjOD79I9CSy0jvdwph52Ktf7utgy1GCYOZGCQ%2Fsentx93eh0Xgl9dME1wwL10mEz2G85rDn6WfyF5xZuZTSttA74glQ%2FwSqzPi4UN%2B%2Bz8UvGNik8sd6PezNI42qdn5FyvSURLrwIZHfB9UOgivEm2kriS6FqTJk96N8Dob%2B1VAIzEoIpPKSONECv21ES5cBZ0ua9JQ94Kimepnz2wBNpU4syKMTcLSk96tmF0pfikEHQGoNAcobkku4Zay%2B7g80C1BPxwjRu1FlRGE4kzJiJ6DIRu6ToO8UtY83kVO01fQPnl4esUsHiP8fuaESnywXxqj%2FUrlaVPW2DYJBsBsXFoJae4Sn8g%2BT%2FGg1TK6mPhWRpeWeelv4%2BGkabiDWuu5mGP8EGJ14KECtbo7eAuxu%2BMxKrrNIqe6ieG5jFHv%2FHaoJa%2FUsAO7riSFJgBxK36Mrx99K6TAIFgvFsYE%2FV5yw2jL2cidZw%2FmFiNXNHQX9erfPHH%2BcJHzePeA7NmKxpfk3Ja2XOwIvT22rHad99UM1X3Svq8qk%2BKeymb0%2FKZj%2Fx7QKOG8o8Ig6daPY6y1TR4bT7DCpWlO8Nh1l7ZVLQ2Nk1v0wjNNq4JLd1fln92Pti1CBAwmRwhHrQH1sY2qHZfhtWJMFzflisKnN%2BJjvPR6Af6kr3z9Hue0HE38TgxQ4u3AWgC%2Fiixgjipe2YWhvE2dPuPC4LHhfj8oRVVPQFhPMzQbNtCk3TfKLjsDMZNDAeaJ9hwnuaWlbMErpgwVtdjHiW4H8qK3H%2FhWZutse23rM5WVyLq1611%2BY49hEAxse3xhxQPmvg%2FliWO7P4Agycfngj%2BK60xun64LtFO6GzrLkwxJKZoAY6sQHxgO8h06r07GQ4yv0xRIaOrm%2BlZ%2BDtcwOFjbC1liDQP%2FQEb7KavY3UJlOXn0K%2BjAnB7kKl1ji1Z8JVLCeVnGsLtaMhyfomZ5LdulnX3qNix7CFuMOkLmtBs0PwlXhnkmxN39PyhM1xKbhPi2bzesPtb%2FhTtYdZmDv%2FPQgfk%2BtHU6zS5FR%2FBIq7sxpVg8lQ%2BVW9JyVgWgxh8IFd2qs9WESjOipV9nCQOhvhuuGqP3Tgojc%3D&Expires=1678263179 [following]\n",
            "--2023-03-06 20:12:59--  https://encode-public.s3.amazonaws.com/2022/01/28/868679f8-fcf5-4906-99b8-4ca0cb1ae34c/ENCFF428QYO.bed.gz?response-content-disposition=attachment%3B%20filename%3DENCFF428QYO.bed.gz&AWSAccessKeyId=ASIATGZNGCNXVHVJY3UO&Signature=Gl9%2FQQhluAZMOipz5nsxs9TdXEo%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEGQaCXVzLXdlc3QtMiJHMEUCIQDNs42kNiu5GM7vFaLtCNH6DYbW4gmTJYbs7HbZtI2qCAIgPoR6U24dak4OrlXiZMnlnzGTAKBFJNtY1WYICAgc4PsqsgUIHRAAGgwyMjA3NDg3MTQ4NjMiDPmqUWnJllxtha8p2CqPBftS7XGYpMYCIHe39YMCLIdoYjOD79I9CSy0jvdwph52Ktf7utgy1GCYOZGCQ%2Fsentx93eh0Xgl9dME1wwL10mEz2G85rDn6WfyF5xZuZTSttA74glQ%2FwSqzPi4UN%2B%2Bz8UvGNik8sd6PezNI42qdn5FyvSURLrwIZHfB9UOgivEm2kriS6FqTJk96N8Dob%2B1VAIzEoIpPKSONECv21ES5cBZ0ua9JQ94Kimepnz2wBNpU4syKMTcLSk96tmF0pfikEHQGoNAcobkku4Zay%2B7g80C1BPxwjRu1FlRGE4kzJiJ6DIRu6ToO8UtY83kVO01fQPnl4esUsHiP8fuaESnywXxqj%2FUrlaVPW2DYJBsBsXFoJae4Sn8g%2BT%2FGg1TK6mPhWRpeWeelv4%2BGkabiDWuu5mGP8EGJ14KECtbo7eAuxu%2BMxKrrNIqe6ieG5jFHv%2FHaoJa%2FUsAO7riSFJgBxK36Mrx99K6TAIFgvFsYE%2FV5yw2jL2cidZw%2FmFiNXNHQX9erfPHH%2BcJHzePeA7NmKxpfk3Ja2XOwIvT22rHad99UM1X3Svq8qk%2BKeymb0%2FKZj%2Fx7QKOG8o8Ig6daPY6y1TR4bT7DCpWlO8Nh1l7ZVLQ2Nk1v0wjNNq4JLd1fln92Pti1CBAwmRwhHrQH1sY2qHZfhtWJMFzflisKnN%2BJjvPR6Af6kr3z9Hue0HE38TgxQ4u3AWgC%2Fiixgjipe2YWhvE2dPuPC4LHhfj8oRVVPQFhPMzQbNtCk3TfKLjsDMZNDAeaJ9hwnuaWlbMErpgwVtdjHiW4H8qK3H%2FhWZutse23rM5WVyLq1611%2BY49hEAxse3xhxQPmvg%2FliWO7P4Agycfngj%2BK60xun64LtFO6GzrLkwxJKZoAY6sQHxgO8h06r07GQ4yv0xRIaOrm%2BlZ%2BDtcwOFjbC1liDQP%2FQEb7KavY3UJlOXn0K%2BjAnB7kKl1ji1Z8JVLCeVnGsLtaMhyfomZ5LdulnX3qNix7CFuMOkLmtBs0PwlXhnkmxN39PyhM1xKbhPi2bzesPtb%2FhTtYdZmDv%2FPQgfk%2BtHU6zS5FR%2FBIq7sxpVg8lQ%2BVW9JyVgWgxh8IFd2qs9WESjOipV9nCQOhvhuuGqP3Tgojc%3D&Expires=1678263179\n",
            "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.218.229.211, 52.92.176.25, 52.218.128.114, ...\n",
            "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.218.229.211|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 223435 (218K) [application/vnd.realvnc.bed]\n",
            "Saving to: ‘ENCFF428QYO.bed.gz’\n",
            "\n",
            "ENCFF428QYO.bed.gz  100%[===================>] 218.20K   544KB/s    in 0.4s    \n",
            "\n",
            "2023-03-06 20:13:01 (544 KB/s) - ‘ENCFF428QYO.bed.gz’ saved [223435/223435]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!wget https://www.encodeproject.org/files/ENCFF428QYO/@@download/ENCFF428QYO.bed.gz\n",
        "!gzip -d ENCFF428QYO.bed.gz"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "MYEGEbQjvEwY"
      },
      "source": [
        "## Сравнение результатов\n",
        "\n",
        "Сравниваем те пики, которые мы получили, с пиками, которые приведены в ENCODE (важно, чтобы версии генома hg38 или hg19) для .bed файла из ENCODE и той хромосомы, которую скачивали выше, сопадали.\n",
        "\n",
        "Проанализируйте полученные результаты и приведите свои рассуждения в README.md. Как можно объяснить различия в количестве пересечений?"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 32,
      "metadata": {
        "id": "zQbSWH-tMtZz"
      },
      "outputs": [],
      "source": [
        "!mkdir venn1\n",
        "!mkdir venn2\n",
        "!mkdir venn3\n",
        "!mkdir venn4"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 72,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "_1EAygSCvc8a",
        "outputId": "66b4a678-2cc3-4751-8c63-ee2bbf561fea"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "Generating a 2-way \"venn\" diagram. Please wait...\n",
            "\n",
            "\n",
            "Done! Please check your results @ venn1. \n",
            "Thank you for using Intervene!\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!intervene venn -i macs2/ENCFF097DJJ_peaks.broadPeak ENCFF428QYO.bed -o venn1 --filenames"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 73,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "7qqn3n2cRdkt",
        "outputId": "791a9f6c-5b4c-46cd-ac12-03980981ab33"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "Generating a 2-way \"venn\" diagram. Please wait...\n",
            "\n",
            "\n",
            "Done! Please check your results @ venn2. \n",
            "Thank you for using Intervene!\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!intervene venn -i ENCFF428QYO.bed macs2/ENCFF097DJJ_peaks.broadPeak -o venn2 --filenames"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 74,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "iZMpe8MBiIoH",
        "outputId": "735725bf-a724-47f8-a437-51164a053904"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "Generating a 2-way \"venn\" diagram. Please wait...\n",
            "\n",
            "\n",
            "Done! Please check your results @ venn3. \n",
            "Thank you for using Intervene!\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!intervene venn -i macs2/ENCFF161DOP_peaks.broadPeak ENCFF428QYO.bed -o venn3 --filenames"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 75,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "lT8If7HXiI9B",
        "outputId": "63580e3a-3b2d-4acb-d7a8-c9b316b4cc28"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "Generating a 2-way \"venn\" diagram. Please wait...\n",
            "\n",
            "\n",
            "Done! Please check your results @ venn4. \n",
            "Thank you for using Intervene!\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!intervene venn -i ENCFF428QYO.bed macs2/ENCFF161DOP_peaks.broadPeak -o venn4 --filenames"
      ]
    },
    {
      "cell_type": "code",
      "source": [],
      "metadata": {
        "id": "PM_7Xi5bg3iJ"
      },
      "execution_count": null,
      "outputs": []
    }
  ],
  "metadata": {
    "colab": {
      "provenance": []
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}